// braid-design-system/dist/styles/lib/components/MenuItemCheckbox/MenuItemCheckbox.css.cjs
"use strict";
const fileScope = require("@vanilla-extract/css/fileScope");
const css = require("@vanilla-extract/css");
const cssUtils = require("@vanilla-extract/css-utils");
const styles_lib_css_responsiveStyle_cjs = require("../../css/responsiveStyle.cjs");
const styles_lib_themes_vars_css_cjs = require("../../themes/vars.css.cjs");
fileScope.setFileScope("src/lib/components/MenuItemCheckbox/MenuItemCheckbox.css.ts?used", "braid-design-system");
const checkboxFieldSize = styles_lib_themes_vars_css_cjs.vars.inlineFieldSize.small;
const menuItemCapHeight = css.createVar("menuItemCapHeight");
const crop = css.createVar("crop");
const checkboxSize = css.style([styles_lib_css_responsiveStyle_cjs.responsiveStyle({
  mobile: {
    vars: {
      [menuItemCapHeight]: styles_lib_themes_vars_css_cjs.vars.textSize.standard.mobile.capHeight,
      [crop]: cssUtils.calc(checkboxFieldSize).subtract(menuItemCapHeight).divide(2).negate().toString()
    }
  },
  tablet: {
    vars: {
      [menuItemCapHeight]: styles_lib_themes_vars_css_cjs.vars.textSize.standard.tablet.capHeight
    }
  }
}), {
  height: checkboxFieldSize,
  width: checkboxFieldSize,
  marginTop: crop,
  marginBottom: crop
}], "checkboxSize");
fileScope.endFileScope();
exports.checkboxSize = checkboxSize;
