// braid-design-system/dist/styles/lib/components/Divider/Divider.css.cjs
"use strict";
const fileScope = require("@vanilla-extract/css/fileScope");
const css = require("@vanilla-extract/css");
const styles_lib_css_colorModeStyle_cjs = require("../../css/colorModeStyle.cjs");
const styles_lib_themes_vars_css_cjs = require("../../themes/vars.css.cjs");
fileScope.setFileScope("src/lib/components/Divider/Divider.css.ts?used", "braid-design-system");
const base = css.style({
  height: styles_lib_themes_vars_css_cjs.vars.borderWidth.standard
}, "base");
const weightVars = css.createThemeContract({
  regular: null,
  strong: null
});
const regular = css.style({
  background: weightVars.regular
}, "regular");
const strong = css.style({
  background: weightVars.strong
}, "strong");
const lightModeVars = {
  vars: css.assignVars(weightVars, {
    regular: styles_lib_themes_vars_css_cjs.vars.borderColor.neutralLight,
    strong: styles_lib_themes_vars_css_cjs.vars.borderColor.neutral
  })
};
const darkModeVars = {
  vars: css.assignVars(weightVars, {
    regular: styles_lib_themes_vars_css_cjs.vars.borderColor.neutral,
    strong: styles_lib_themes_vars_css_cjs.vars.borderColor.neutralLight
  })
};
const lightModeWeight = css.styleVariants({
  light: styles_lib_css_colorModeStyle_cjs.colorModeStyle({
    lightMode: lightModeVars
  }),
  dark: styles_lib_css_colorModeStyle_cjs.colorModeStyle({
    lightMode: darkModeVars
  })
}, "lightModeWeight");
const darkModeWeight = css.styleVariants({
  light: styles_lib_css_colorModeStyle_cjs.colorModeStyle({
    darkMode: lightModeVars
  }),
  dark: styles_lib_css_colorModeStyle_cjs.colorModeStyle({
    darkMode: darkModeVars
  })
}, "darkModeWeight");
fileScope.endFileScope();
exports.base = base;
exports.darkModeWeight = darkModeWeight;
exports.lightModeWeight = lightModeWeight;
exports.regular = regular;
exports.strong = strong;
