// braid-design-system/dist/styles/lib/css/reset/reset.css.mjs
import { setFileScope, endFileScope } from "@vanilla-extract/css/fileScope";
import { style } from "@vanilla-extract/css";
import { hideFocusRingsDataAttribute } from "../../components/private/hideFocusRings/hideFocusRingsDataAttribute.mjs";
setFileScope("src/lib/css/reset/reset.css.ts?used", "braid-design-system");
const base = style({
  margin: 0,
  padding: 0,
  border: 0,
  boxSizing: "border-box",
  fontSize: "100%",
  font: "inherit",
  verticalAlign: "baseline",
  WebkitTapHighlightColor: "transparent",
  selectors: {
    [`[${hideFocusRingsDataAttribute}] &`]: {
      outline: "none"
    }
  }
}, "base");
const block = style({
  display: "block"
}, "block");
const body = style({
  lineHeight: 1
}, "body");
const list = style({
  listStyle: "none"
}, "list");
const quote = style({
  quotes: "none",
  selectors: {
    "&:before, &:after": {
      content: "''"
    }
  }
}, "quote");
const table = style({
  borderCollapse: "collapse",
  borderSpacing: 0
}, "table");
const appearance = style({
  appearance: "none"
}, "appearance");
const transparent = style({
  backgroundColor: "transparent"
}, "transparent");
const field = style([block, appearance, transparent], "field");
const mark = style([transparent, {
  color: "inherit"
}], "mark");
const select = style([field, style({
  ":disabled": {
    opacity: 1
  },
  selectors: {
    "&::-ms-expand": {
      display: "none"
    }
  }
}, "select")], "select");
const input = style([field, style({
  selectors: {
    ['&[type="number"]']: {
      MozAppearance: "textfield"
    },
    [['&[type="number"]::-webkit-inner-spin-button', '&[type="number"]::-webkit-outer-spin-button'].join(",")]: {
      WebkitAppearance: "none",
      margin: 0
    },
    "&::-ms-clear": {
      display: "none"
    },
    "&::-webkit-search-cancel-button": {
      WebkitAppearance: "none"
    }
  }
}, "input")], "input");
const button = style([transparent], "button");
const a = style({
  textDecoration: "none",
  color: "inherit"
}, "a");
const element = {
  article: block,
  aside: block,
  details: block,
  figcaption: block,
  figure: block,
  footer: block,
  header: block,
  hgroup: block,
  menu: block,
  nav: block,
  section: block,
  ul: list,
  ol: list,
  blockquote: quote,
  q: quote,
  body,
  a,
  table,
  mark,
  select,
  button,
  textarea: field,
  input
};
endFileScope();
export {
  base,
  element
};
