// braid-design-system/dist/styles/lib/css/reset/reset.css.cjs
"use strict";
const fileScope = require("@vanilla-extract/css/fileScope");
const css = require("@vanilla-extract/css");
const styles_lib_components_private_hideFocusRings_hideFocusRingsDataAttribute_cjs = require("../../components/private/hideFocusRings/hideFocusRingsDataAttribute.cjs");
fileScope.setFileScope("src/lib/css/reset/reset.css.ts?used", "braid-design-system");
const base = css.style({
  margin: 0,
  padding: 0,
  border: 0,
  boxSizing: "border-box",
  fontSize: "100%",
  font: "inherit",
  verticalAlign: "baseline",
  WebkitTapHighlightColor: "transparent",
  selectors: {
    [`[${styles_lib_components_private_hideFocusRings_hideFocusRingsDataAttribute_cjs.hideFocusRingsDataAttribute}] &`]: {
      outline: "none"
    }
  }
}, "base");
const block = css.style({
  display: "block"
}, "block");
const body = css.style({
  lineHeight: 1
}, "body");
const list = css.style({
  listStyle: "none"
}, "list");
const quote = css.style({
  quotes: "none",
  selectors: {
    "&:before, &:after": {
      content: "''"
    }
  }
}, "quote");
const table = css.style({
  borderCollapse: "collapse",
  borderSpacing: 0
}, "table");
const appearance = css.style({
  appearance: "none"
}, "appearance");
const transparent = css.style({
  backgroundColor: "transparent"
}, "transparent");
const field = css.style([block, appearance, transparent], "field");
const mark = css.style([transparent, {
  color: "inherit"
}], "mark");
const select = css.style([field, css.style({
  ":disabled": {
    opacity: 1
  },
  selectors: {
    "&::-ms-expand": {
      display: "none"
    }
  }
}, "select")], "select");
const input = css.style([field, css.style({
  selectors: {
    ['&[type="number"]']: {
      MozAppearance: "textfield"
    },
    [['&[type="number"]::-webkit-inner-spin-button', '&[type="number"]::-webkit-outer-spin-button'].join(",")]: {
      WebkitAppearance: "none",
      margin: 0
    },
    "&::-ms-clear": {
      display: "none"
    },
    "&::-webkit-search-cancel-button": {
      WebkitAppearance: "none"
    }
  }
}, "input")], "input");
const button = css.style([transparent], "button");
const a = css.style({
  textDecoration: "none",
  color: "inherit"
}, "a");
const element = {
  article: block,
  aside: block,
  details: block,
  figcaption: block,
  figure: block,
  footer: block,
  header: block,
  hgroup: block,
  menu: block,
  nav: block,
  section: block,
  ul: list,
  ol: list,
  blockquote: quote,
  q: quote,
  body,
  a,
  table,
  mark,
  select,
  button,
  textarea: field,
  input
};
fileScope.endFileScope();
exports.base = base;
exports.element = element;
