// braid-design-system/dist/styles/lib/components/Textarea/Highlight/Highlight.css.cjs
"use strict";
const fileScope = require("@vanilla-extract/css/fileScope");
const css = require("@vanilla-extract/css");
const styles_lib_css_atoms_atoms_cjs = require("../../../css/atoms/atoms.cjs");
const styles_lib_css_colorModeStyle_cjs = require("../../../css/colorModeStyle.cjs");
const styles_lib_themes_vars_css_cjs = require("../../../themes/vars.css.cjs");
fileScope.setFileScope("src/lib/components/Textarea/Highlight/Highlight.css.ts?used", "braid-design-system");
const space = "2px";
const root = css.style([styles_lib_css_atoms_atoms_cjs.atoms({
  borderRadius: "small"
}), {
  padding: `0 ${space}`,
  margin: `0 -${space}`,
  textDecoration: "underline",
  textDecorationStyle: "wavy",
  textDecorationSkipInk: "none",
  textDecorationThickness: 2,
  textUnderlineOffset: 2
}], "root");
const critical = css.style(styles_lib_css_colorModeStyle_cjs.colorModeStyle({
  lightMode: {
    textDecorationColor: styles_lib_themes_vars_css_cjs.vars.borderColor.critical,
    background: styles_lib_themes_vars_css_cjs.vars.backgroundColor.criticalLight
  },
  darkMode: {
    textDecorationColor: styles_lib_themes_vars_css_cjs.vars.borderColor.criticalLight
  }
}), "critical");
const caution = css.style([{
  textDecorationColor: styles_lib_themes_vars_css_cjs.vars.borderColor.caution
}, styles_lib_css_colorModeStyle_cjs.colorModeStyle({
  lightMode: {
    background: styles_lib_themes_vars_css_cjs.vars.backgroundColor.cautionLight
  }
})], "caution");
fileScope.endFileScope();
exports.caution = caution;
exports.critical = critical;
exports.root = root;
