// braid-design-system/dist/styles/lib/themes/seekJobs/tokens.cjs
"use strict";
const polished = require("polished");
const sideEffects_lib_themes_baseTokens_apac_cjs = require("../../../../side-effects/lib/themes/baseTokens/apac.cjs");
const brandAccent = sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekPink["500"];
const brandAccentSoft = sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekPink["50"];
const tokens = sideEffects_lib_themes_baseTokens_apac_cjs.makeTokens({
  name: "seekJobs",
  displayName: "SEEK Jobs",
  brand: sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlue["700"],
  brandAccent,
  brandAccentLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekPink["200"],
  brandAccentActive: polished.darken(0.05, brandAccent),
  brandAccentHover: polished.lighten(0.05, brandAccent),
  brandAccentSoft,
  brandAccentSoftActive: polished.darken(0.05, brandAccentSoft),
  brandAccentSoftHover: polished.darken(0.025, brandAccentSoft),
  tokenOverrides: {
    legacy: false,
    typography: {
      fontFamily: 'SeekSans, "SeekSans Fallback", Arial',
      webFont: "https://www.seek.com.au/static/shared-web/seeksans.css",
      fontMetrics: {
        capHeight: 783,
        ascent: 1057,
        descent: -274,
        lineGap: 0,
        unitsPerEm: 1e3
      },
      fontWeight: {
        regular: 400,
        medium: 500,
        strong: 700
      },
      heading: {
        weight: {
          weak: "regular",
          regular: "medium"
        },
        level: {
          "1": {
            mobile: {
              fontSize: 28,
              lineGap: 11
            },
            tablet: {
              fontSize: 36,
              lineGap: 14
            }
          },
          "2": {
            mobile: {
              fontSize: 24,
              lineGap: 11
            },
            tablet: {
              fontSize: 30,
              lineGap: 13
            }
          },
          "3": {
            mobile: {
              fontSize: 22,
              lineGap: 10
            },
            tablet: {
              fontSize: 24,
              lineGap: 11
            }
          },
          "4": {
            mobile: {
              fontSize: 20,
              lineGap: 9
            },
            tablet: {
              fontSize: 20,
              lineGap: 9
            }
          }
        }
      },
      text: {
        large: {
          mobile: {
            fontSize: 18,
            lineGap: 13
          },
          tablet: {
            fontSize: 18,
            lineGap: 13
          }
        },
        standard: {
          mobile: {
            fontSize: 16,
            lineGap: 12
          },
          tablet: {
            fontSize: 16,
            lineGap: 12
          }
        },
        small: {
          mobile: {
            fontSize: 14,
            lineGap: 10
          },
          tablet: {
            fontSize: 14,
            lineGap: 10
          }
        },
        xsmall: {
          mobile: {
            fontSize: 12,
            lineGap: 9
          },
          tablet: {
            fontSize: 12,
            lineGap: 9
          }
        }
      }
    },
    space: {
      gutter: 8,
      xxsmall: 2,
      xsmall: 3,
      small: 4,
      medium: 6,
      large: 8,
      xlarge: 12,
      xxlarge: 16,
      xxxlarge: 24
    },
    focusRingSize: 6,
    touchableSize: 12,
    border: {
      width: {
        standard: 2,
        large: 4
      },
      radius: {
        small: "4px",
        standard: "8px",
        large: "16px",
        xlarge: "24px"
      },
      color: {
        formAccent: sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["700"],
        formAccentLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["300"],
        focus: polished.rgba(sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["300"], 0.7),
        promote: sideEffects_lib_themes_baseTokens_apac_cjs.palette.purple["700"],
        promoteLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.purple["300"],
        neutralLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["100"]
      }
    },
    color: {
      foreground: {
        formAccent: sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["700"],
        formAccentLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["300"],
        link: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["700"],
        linkHover: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["700"],
        linkLight: "#fff",
        linkVisited: sideEffects_lib_themes_baseTokens_apac_cjs.palette.purple["800"],
        linkLightVisited: sideEffects_lib_themes_baseTokens_apac_cjs.palette.purple["200"],
        promote: sideEffects_lib_themes_baseTokens_apac_cjs.palette.purple["700"],
        promoteLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.purple["300"],
        rating: sideEffects_lib_themes_baseTokens_apac_cjs.palette.orange["600"]
      },
      background: {
        body: "#fff",
        bodyDark: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["800"],
        formAccent: sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["700"],
        formAccentActive: polished.darken(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["700"]),
        formAccentHover: polished.lighten(0.075, sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["700"]),
        formAccentSoft: sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["50"],
        formAccentSoftActive: polished.darken(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["50"]),
        formAccentSoftHover: polished.darken(0.025, sideEffects_lib_themes_baseTokens_apac_cjs.palette.seekBlueLight["50"]),
        promote: sideEffects_lib_themes_baseTokens_apac_cjs.palette.purple["700"],
        promoteLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.purple["100"],
        neutralLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["75"],
        neutralSoft: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["75"],
        neutralSoftActive: polished.darken(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["75"]),
        neutralSoftHover: polished.darken(0.025, sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["75"])
      }
    }
  }
});
exports.tokens = tokens;
