// braid-design-system/dist/styles/lib/components/Pagination/Pagination.css.cjs
"use strict";
const fileScope = require("@vanilla-extract/css/fileScope");
const css = require("@vanilla-extract/css");
const styles_lib_css_colorModeStyle_cjs = require("../../css/colorModeStyle.cjs");
fileScope.setFileScope("src/lib/components/Pagination/Pagination.css.ts?used", "braid-design-system");
const hover = css.style({}, "hover");
const lightModeCurrentKeyline = css.style(styles_lib_css_colorModeStyle_cjs.colorModeStyle({
  lightMode: {
    opacity: 0.3
  }
}), "lightModeCurrentKeyline");
const darkModeCurrentKeyline = css.style(styles_lib_css_colorModeStyle_cjs.colorModeStyle({
  darkMode: {
    opacity: 0.3
  }
}), "darkModeCurrentKeyline");
const current = css.style({
  opacity: 0.075
}, "current");
const background = css.style({
  selectors: {
    [`${hover}:hover &:not(${current})`]: {
      opacity: 0.5
    }
  }
}, "background");
fileScope.endFileScope();
exports.background = background;
exports.current = current;
exports.darkModeCurrentKeyline = darkModeCurrentKeyline;
exports.hover = hover;
exports.lightModeCurrentKeyline = lightModeCurrentKeyline;
