// braid-design-system/dist/styles/lib/components/TextDropdown/TextDropdown.css.cjs
"use strict";
const fileScope = require("@vanilla-extract/css/fileScope");
const css = require("@vanilla-extract/css");
const cssUtils = require("@vanilla-extract/css-utils");
const styles_lib_themes_vars_css_cjs = require("../../themes/vars.css.cjs");
const styles_lib_components_private_touchable_virtualTouchableRules_cjs = require("../private/touchable/virtualTouchableRules.cjs");
fileScope.setFileScope("src/lib/components/TextDropdown/TextDropdown.css.ts?used", "braid-design-system");
const select = css.style({
  ...styles_lib_components_private_touchable_virtualTouchableRules_cjs.virtualTouchableRules
}, "select");
const focusOverlay = [css.style({
  selectors: {
    [`${select}:focus ~ &`]: {
      opacity: 1
    }
  }
}, "focusOverlay"), css.style({
  top: cssUtils.calc.negate(styles_lib_themes_vars_css_cjs.vars.space.xxsmall),
  bottom: cssUtils.calc.negate(styles_lib_themes_vars_css_cjs.vars.space.xxsmall),
  left: cssUtils.calc.negate(styles_lib_themes_vars_css_cjs.vars.space.xxsmall),
  right: cssUtils.calc.negate(styles_lib_themes_vars_css_cjs.vars.space.xxsmall)
}, "focusOverlay")];
fileScope.endFileScope();
exports.focusOverlay = focusOverlay;
exports.select = select;
