// braid-design-system/dist/styles/lib/themes/docs/tokens.cjs
"use strict";
const appleSystemMetrics = require("@capsizecss/metrics/appleSystem.js");
const polished = require("polished");
const sideEffects_lib_themes_baseTokens_apac_cjs = require("../../../../side-effects/lib/themes/baseTokens/apac.cjs");
const sideEffects_lib_themes_tokenType_cjs = require("../../../../side-effects/lib/themes/tokenType.cjs");
const _interopDefaultCompat = (e) => e && typeof e === "object" && "default" in e ? e : { default: e };
const appleSystemMetrics__default = /* @__PURE__ */ _interopDefaultCompat(appleSystemMetrics);
const brandAccent = sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["900"];
const formAccent = sideEffects_lib_themes_baseTokens_apac_cjs.palette.indigo["600"];
const focus = polished.rgba("#1e90ff", 0.7);
const black = sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["700"];
const white = "#fff";
const link = sideEffects_lib_themes_baseTokens_apac_cjs.palette.indigo["600"];
const linkVisited = "DarkViolet";
const secondary = sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["500"];
const tokens = {
  name: "docs",
  displayName: "Docs",
  legacy: false,
  typography: {
    fontFamily: '-apple-system, BlinkMacSystemFont, Roboto, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"',
    webFont: null,
    fontMetrics: sideEffects_lib_themes_tokenType_cjs.extractFontMetricsForTheme(appleSystemMetrics__default.default),
    fontWeight: {
      regular: 400,
      medium: 500,
      strong: 700
    },
    heading: {
      weight: {
        weak: "regular",
        regular: "strong"
      },
      level: {
        "1": {
          mobile: {
            fontSize: 32,
            rows: 10
          },
          tablet: {
            fontSize: 52,
            rows: 15
          }
        },
        "2": {
          mobile: {
            fontSize: 28,
            rows: 9
          },
          tablet: {
            fontSize: 38,
            rows: 13
          }
        },
        "3": {
          mobile: {
            fontSize: 22,
            rows: 8
          },
          tablet: {
            fontSize: 22,
            rows: 8
          }
        },
        "4": {
          mobile: {
            fontSize: 18,
            rows: 8
          },
          tablet: {
            fontSize: 18,
            rows: 8
          }
        }
      }
    },
    text: {
      xsmall: {
        mobile: {
          fontSize: 14,
          rows: 6
        },
        tablet: {
          fontSize: 14,
          rows: 6
        }
      },
      small: {
        mobile: {
          fontSize: 16,
          rows: 6
        },
        tablet: {
          fontSize: 16,
          rows: 6
        }
      },
      standard: {
        mobile: {
          fontSize: 18,
          rows: 8
        },
        tablet: {
          fontSize: 18,
          rows: 8
        }
      },
      large: {
        mobile: {
          fontSize: 22,
          rows: 9
        },
        tablet: {
          fontSize: 22,
          rows: 9
        }
      }
    }
  },
  contentWidth: {
    xsmall: 400,
    small: 660,
    medium: 940,
    large: 1308
  },
  grid: 4,
  touchableSize: 12,
  space: {
    gutter: 6,
    xxsmall: 1,
    xsmall: 2,
    small: 3,
    medium: 5,
    large: 8,
    xlarge: 11,
    xxlarge: 15,
    xxxlarge: 20
  },
  transforms: {
    touchable: "scale(0.97)"
  },
  transitions: {
    fast: "transform .125s ease, opacity .125s ease",
    touchable: "transform 0.2s cubic-bezier(0.02, 1.505, 0.745, 1.235)"
  },
  border: {
    radius: {
      small: "4px",
      standard: "6px",
      large: "8px",
      xlarge: "12px"
    },
    width: {
      standard: 1,
      large: 2
    },
    color: {
      brandAccent,
      brandAccentLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["200"],
      caution: sideEffects_lib_themes_baseTokens_apac_cjs.palette.yellow["600"],
      cautionLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.yellow["300"],
      critical: sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["600"],
      criticalLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["400"],
      field: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["400"],
      focus,
      formAccent,
      formAccentLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.indigo["400"],
      info: sideEffects_lib_themes_baseTokens_apac_cjs.palette.blue["600"],
      infoLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.blue["400"],
      neutral: black,
      neutralInverted: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["100"],
      neutralLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["200"],
      positive: sideEffects_lib_themes_baseTokens_apac_cjs.palette.mint["600"],
      positiveLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.mint["400"],
      promote: sideEffects_lib_themes_baseTokens_apac_cjs.palette.violet["600"],
      promoteLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.violet["400"]
    }
  },
  focusRingSize: 3,
  shadows: {
    small: "0 2px 4px 0px rgba(28,28,28,.1), 0 2px 2px -2px rgba(28,28,28,.1), 0 4px 4px -4px rgba(28,28,28,.2)",
    medium: "0 2px 4px 0px rgba(28,28,28,.1), 0 8px 8px -4px rgba(28,28,28,.1), 0 12px 12px -8px rgba(28,28,28,.2)",
    large: "0 2px 4px 0px rgba(28,28,28,.1), 0 12px 12px -4px rgba(28,28,28,.1), 0 20px 20px -12px rgba(28,28,28,.2)"
  },
  color: {
    foreground: {
      brandAccent,
      brandAccentLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["200"],
      caution: sideEffects_lib_themes_baseTokens_apac_cjs.palette.yellow["900"],
      cautionLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.yellow["300"],
      critical: sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["700"],
      criticalLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["400"],
      formAccent,
      formAccentLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.indigo["400"],
      info: sideEffects_lib_themes_baseTokens_apac_cjs.palette.blue["700"],
      infoLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.blue["400"],
      link,
      linkHover: link,
      linkLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.indigo["300"],
      linkVisited,
      linkLightVisited: linkVisited,
      neutral: black,
      neutralInverted: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["100"],
      positive: sideEffects_lib_themes_baseTokens_apac_cjs.palette.mint["700"],
      positiveLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.mint["400"],
      promote: sideEffects_lib_themes_baseTokens_apac_cjs.palette.violet["700"],
      promoteLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.violet["400"],
      rating: black,
      secondary,
      secondaryInverted: polished.rgba("#fff", 0.65)
    },
    background: {
      body: white,
      bodyDark: polished.darken(0.04, sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["900"]),
      brand: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["900"],
      brandAccent,
      brandAccentActive: brandAccent,
      brandAccentHover: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["700"],
      brandAccentSoft: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["100"],
      brandAccentSoftActive: polished.darken(0.025, sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["100"]),
      brandAccentSoftHover: polished.darken(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["100"]),
      caution: sideEffects_lib_themes_baseTokens_apac_cjs.palette.yellow["500"],
      cautionLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.yellow["100"],
      critical: sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["600"],
      criticalActive: polished.darken(0.075, sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["600"]),
      criticalHover: polished.darken(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["600"]),
      criticalLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["100"],
      criticalSoft: sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["100"],
      criticalSoftActive: polished.darken(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["100"]),
      criticalSoftHover: polished.darken(0.025, sideEffects_lib_themes_baseTokens_apac_cjs.palette.red["100"]),
      formAccent,
      formAccentActive: sideEffects_lib_themes_baseTokens_apac_cjs.palette.indigo["700"],
      formAccentHover: polished.darken(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.indigo["600"]),
      formAccentSoft: sideEffects_lib_themes_baseTokens_apac_cjs.palette.indigo["50"],
      formAccentSoftActive: polished.darken(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.indigo["50"]),
      formAccentSoftHover: polished.darken(0.025, sideEffects_lib_themes_baseTokens_apac_cjs.palette.indigo["50"]),
      info: sideEffects_lib_themes_baseTokens_apac_cjs.palette.blue["600"],
      infoLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.blue["100"],
      neutral: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["800"],
      neutralActive: polished.darken(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["800"]),
      neutralHover: polished.lighten(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["800"]),
      neutralLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["100"],
      neutralSoft: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["50"],
      neutralSoftActive: polished.darken(0.05, sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["50"]),
      neutralSoftHover: polished.darken(0.025, sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["50"]),
      positive: sideEffects_lib_themes_baseTokens_apac_cjs.palette.mint["700"],
      positiveLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.mint["100"],
      promote: sideEffects_lib_themes_baseTokens_apac_cjs.palette.violet["600"],
      promoteLight: sideEffects_lib_themes_baseTokens_apac_cjs.palette.violet["100"],
      surface: white,
      surfaceDark: sideEffects_lib_themes_baseTokens_apac_cjs.palette.grey["900"]
    }
  }
};
exports.tokens = tokens;
