// braid-design-system/dist/styles/lib/components/Rating/Rating.css.cjs
"use strict";
const fileScope = require("@vanilla-extract/css/fileScope");
const css = require("@vanilla-extract/css");
const styles_lib_css_colorModeStyle_cjs = require("../../css/colorModeStyle.cjs");
const styles_lib_themes_vars_css_cjs = require("../../themes/vars.css.cjs");
fileScope.setFileScope("src/lib/components/Rating/Rating.css.ts?used", "braid-design-system");
const lightModeStarColor = css.style(styles_lib_css_colorModeStyle_cjs.colorModeStyle({
  lightMode: {
    color: styles_lib_themes_vars_css_cjs.vars.foregroundColor.rating
  }
}), "lightModeStarColor");
const darkModeStarColor = css.style(styles_lib_css_colorModeStyle_cjs.colorModeStyle({
  darkMode: {
    color: styles_lib_themes_vars_css_cjs.vars.foregroundColor.rating
  }
}), "darkModeStarColor");
const starSpacing = css.style({
  paddingRight: "1px"
}, "starSpacing");
const textSpacing = css.style({
  paddingLeft: "0.4em"
}, "textSpacing");
fileScope.endFileScope();
exports.darkModeStarColor = darkModeStarColor;
exports.lightModeStarColor = lightModeStarColor;
exports.starSpacing = starSpacing;
exports.textSpacing = textSpacing;
