// braid-design-system/dist/styles/lib/css/atoms/atomicProperties.cjs
"use strict";
const styles_lib_themes_vars_css_cjs = require("../../themes/vars.css.cjs");
const sizes = {
  full: "100%",
  touchable: styles_lib_themes_vars_css_cjs.vars.touchableSize
};
const space = {
  ...styles_lib_themes_vars_css_cjs.vars.space,
  none: 0
};
const boxShadow = {
  ...styles_lib_themes_vars_css_cjs.vars.shadow,
  borderBrandAccent: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.brandAccent}`,
  borderBrandAccentLight: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.brandAccentLight}`,
  borderBrandAccentLarge: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.large} ${styles_lib_themes_vars_css_cjs.vars.borderColor.brandAccent}`,
  borderBrandAccentLightLarge: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.large} ${styles_lib_themes_vars_css_cjs.vars.borderColor.brandAccentLight}`,
  borderCaution: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.caution}`,
  borderCautionLight: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.cautionLight}`,
  borderCritical: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.critical}`,
  borderCriticalLarge: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.large} ${styles_lib_themes_vars_css_cjs.vars.borderColor.critical}`,
  borderCriticalLight: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.criticalLight}`,
  borderCriticalLightLarge: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.large} ${styles_lib_themes_vars_css_cjs.vars.borderColor.criticalLight}`,
  borderField: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.field}`,
  borderFormAccent: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.formAccent}`,
  borderFormAccentLarge: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.large} ${styles_lib_themes_vars_css_cjs.vars.borderColor.formAccent}`,
  borderFormAccentLight: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.formAccentLight}`,
  borderFormAccentLightLarge: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.large} ${styles_lib_themes_vars_css_cjs.vars.borderColor.formAccentLight}`,
  borderInfo: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.info}`,
  borderInfoLight: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.infoLight}`,
  borderNeutral: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.neutral}`,
  borderNeutralLarge: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.large} ${styles_lib_themes_vars_css_cjs.vars.borderColor.neutral}`,
  borderNeutralInverted: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.neutralInverted}`,
  borderNeutralInvertedLarge: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.large} ${styles_lib_themes_vars_css_cjs.vars.borderColor.neutralInverted}`,
  borderNeutralLight: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.neutralLight}`,
  borderPositive: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.positive}`,
  borderPositiveLight: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.positiveLight}`,
  borderPromote: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.promote}`,
  borderPromoteLight: `inset 0 0 0 ${styles_lib_themes_vars_css_cjs.vars.borderWidth.standard} ${styles_lib_themes_vars_css_cjs.vars.borderColor.promoteLight}`,
  outlineFocus: `0 0 0 ${styles_lib_themes_vars_css_cjs.vars.focusRingSize} ${styles_lib_themes_vars_css_cjs.vars.borderColor.focus}`
};
const pseudoProperties = {
  transform: styles_lib_themes_vars_css_cjs.vars.transform
};
const unresponsiveProperties = {
  overflow: ["hidden", "scroll", "visible", "auto"],
  userSelect: ["none"],
  outline: ["none"],
  opacity: [0],
  zIndex: {
    0: 0,
    1: 1,
    2: 2,
    dropdownBackdrop: 90,
    dropdown: 100,
    sticky: 200,
    modalBackdrop: 290,
    modal: 300,
    notification: 400
  },
  cursor: ["default", "pointer"],
  pointerEvents: ["none"],
  top: [0],
  bottom: [0],
  left: [0],
  right: [0],
  height: sizes,
  width: sizes,
  minWidth: {
    0: "0%"
  },
  maxWidth: styles_lib_themes_vars_css_cjs.vars.contentWidth,
  transition: styles_lib_themes_vars_css_cjs.vars.transition
};
const colorProperties = {
  background: styles_lib_themes_vars_css_cjs.vars.backgroundColor,
  boxShadow
};
const responsiveProperties = {
  display: {
    none: "none",
    block: "block",
    inline: "inline",
    inlineBlock: "inline-block",
    flex: "flex"
  },
  position: ["relative", "absolute", "fixed"],
  borderRadius: {
    none: "0px",
    full: "9999px",
    ...styles_lib_themes_vars_css_cjs.vars.borderRadius
  },
  paddingTop: space,
  paddingBottom: space,
  paddingRight: space,
  paddingLeft: space,
  marginTop: space,
  marginBottom: space,
  marginRight: space,
  marginLeft: space,
  alignItems: {
    flexStart: "flex-start",
    center: "center",
    flexEnd: "flex-end"
  },
  justifyContent: {
    flexStart: "flex-start",
    center: "center",
    flexEnd: "flex-end",
    spaceBetween: "space-between"
  },
  flexDirection: {
    row: "row",
    rowReverse: "row-reverse",
    column: "column",
    columnReverse: "column-reverse"
  },
  flexWrap: {
    wrap: "wrap",
    nowrap: "nowrap"
  },
  flexShrink: [0],
  flexGrow: [0, 1],
  textAlign: ["left", "center", "right"]
};
exports.colorProperties = colorProperties;
exports.pseudoProperties = pseudoProperties;
exports.responsiveProperties = responsiveProperties;
exports.space = space;
exports.unresponsiveProperties = unresponsiveProperties;
