// braid-design-system/dist/ToastContext.chunk.cjs
"use strict";
const React = require("react");
const sideEffects_lib_components_BraidProvider_BraidProvider_cjs = require("./side-effects/lib/components/BraidProvider/BraidProvider.cjs");
const styles_lib_css_breakpoints_cjs = require("./styles/lib/css/breakpoints.cjs");
const jsxRuntime = require("react/jsx-runtime");
const reactDom = require("react-dom");
const clsx = require("clsx");
const dedent = require("dedent");
const styles_lib_css_reset_reset_css_cjs = require("./styles/lib/css/reset/reset.css.cjs");
const styles_lib_css_atoms_atoms_cjs = require("./styles/lib/css/atoms/atoms.cjs");
const styles_lib_css_atoms_sprinkles_css_cjs = require("./styles/lib/css/atoms/sprinkles.css.cjs");
const styles_lib_css_typography_css_cjs = require("./styles/lib/css/typography.css.cjs");
const assert = require("assert");
const reactIs = require("react-is");
const styles_lib_components_Divider_Divider_css_cjs = require("./styles/lib/components/Divider/Divider.css.cjs");
const styles_lib_components_Hidden_Hidden_css_cjs = require("./styles/lib/components/Hidden/Hidden.css.cjs");
const styles_lib_css_negativeMargin_negativeMargin_css_cjs = require("./styles/lib/css/negativeMargin/negativeMargin.css.cjs");
const styles_lib_components_Column_Column_css_cjs = require("./styles/lib/components/Column/Column.css.cjs");
const styles_lib_components_ContentBlock_ContentBlock_css_cjs = require("./styles/lib/components/ContentBlock/ContentBlock.css.cjs");
const dynamic = require("@vanilla-extract/dynamic");
const styles_lib_components_private_MaxLines_MaxLines_css_cjs = require("./styles/lib/components/private/MaxLines/MaxLines.css.cjs");
const styles_lib_components_private_AvoidWidowIcon_AvoidWidowIcon_css_cjs = require("./styles/lib/components/private/AvoidWidowIcon/AvoidWidowIcon.css.cjs");
const styles_lib_components_private_touchable_virtualTouchable_css_cjs = require("./styles/lib/components/private/touchable/virtualTouchable.css.cjs");
const styles_lib_components_TextLink_TextLink_css_cjs = require("./styles/lib/components/TextLink/TextLink.css.cjs");
const styles_lib_css_lineHeightContainer_css_cjs = require("./styles/lib/css/lineHeightContainer.css.cjs");
const styles_lib_hooks_useIcon_icon_css_cjs = require("./styles/lib/hooks/useIcon/icon.css.cjs");
const styles_lib_components_private_hideFocusRings_hideFocusRings_css_cjs = require("./styles/lib/components/private/hideFocusRings/hideFocusRings.css.cjs");
const styles_lib_components_Button_Button_css_cjs = require("./styles/lib/components/Button/Button.css.cjs");
const reactPopperTooltip = require("react-popper-tooltip");
const isMobile = require("is-mobile");
const styles_lib_components_TooltipRenderer_TooltipRenderer_css_cjs = require("./styles/lib/components/TooltipRenderer/TooltipRenderer.css.cjs");
const styles_lib_components_ButtonIcon_ButtonIcon_css_cjs = require("./styles/lib/components/ButtonIcon/ButtonIcon.css.cjs");
const styles_lib_components_useToast_Toast_css_cjs = require("./styles/lib/components/useToast/Toast.css.cjs");
const _interopDefaultCompat = (e) => e && typeof e === "object" && "default" in e ? e : { default: e };
const React__default = /* @__PURE__ */ _interopDefaultCompat(React);
const clsx__default = /* @__PURE__ */ _interopDefaultCompat(clsx);
const dedent__default = /* @__PURE__ */ _interopDefaultCompat(dedent);
const assert__default = /* @__PURE__ */ _interopDefaultCompat(assert);
const isMobile__default = /* @__PURE__ */ _interopDefaultCompat(isMobile);
const TextContext = React.createContext(null);
const BraidPortal = ({ children, container }) => {
  const { vanillaTheme } = sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useBraidTheme();
  return reactDom.createPortal(
    /* @__PURE__ */ jsxRuntime.jsx(TextContext.Provider, { value: null, children: /* @__PURE__ */ jsxRuntime.jsx(sideEffects_lib_components_BraidProvider_BraidProvider_cjs.VanillaThemeContainer, { theme: vanillaTheme, setDefaultTextTones: true, children }) }),
    container ?? document.body
  );
};
const useThemeName = () => sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useBraidTheme().name;
const useSpace = () => sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useBraidTheme().space;
function resolveResponsiveObject(breakpointName, value) {
  let resolvedValue = null;
  for (const currentBreakpointName of styles_lib_css_breakpoints_cjs.breakpointNames) {
    const valueForBreakpoint = value[currentBreakpointName];
    if (valueForBreakpoint !== void 0) {
      resolvedValue = valueForBreakpoint;
    }
    if (breakpointName === currentBreakpointName) {
      return resolvedValue;
    }
  }
  return resolvedValue;
}
const useResponsiveValue = () => {
  const breakpointName = React.useContext(sideEffects_lib_components_BraidProvider_BraidProvider_cjs.breakpointContext);
  return function responsiveValue(value) {
    return breakpointName ? resolveResponsiveObject(breakpointName, value) : null;
  };
};
const buildDataAttributes = ({ data, validateRestProps }) => {
  if (process.env.NODE_ENV !== "production") {
    if (validateRestProps !== false) {
      const dataAttrs = Object.keys(validateRestProps).filter(
        (prop) => prop.startsWith("data-")
      );
      const formatter = (v) => {
        if (typeof v === "boolean") {
          return v ? "" : `={false}`;
        }
        if (typeof v === "number") {
          return `={${v}}`;
        }
        return `="${v}"`;
      };
      if (dataAttrs.length > 0) {
        console.warn(
          dedent__default.default`
          Braid components do not support the native data attribute format. Use the “data” prop instead.
            <Component
          %c${dataAttrs.map((attr) => `-    ${attr}${formatter(validateRestProps[attr])}`).join("\n")}
          %c+    data={{
          ${dataAttrs.map((attr) => {
            const attributeName = attr.replace(/^data-/, "");
            const property = attributeName.includes("-") ? `'${attributeName}'` : attributeName;
            return `+      ${property}: ${typeof validateRestProps[attr] === "string" ? `'${validateRestProps[attr]}'` : validateRestProps[attr]},`;
          }).join("\n")}
          +    }}
            %c/>
           For more details, see the “Data Attributes” documentation:
           https://seek-oss.github.io/braid-design-system/components/Box#data-attributes
        `,
          "color: red",
          "color: green",
          "color: inherit"
        );
      }
    }
  }
  if (!data) {
    return;
  }
  const keys = Object.keys(data);
  const dataAttributes = {};
  for (const key of keys) {
    dataAttributes[`data-${key}`] = data[key];
  }
  return dataAttributes;
};
const backgroundContext = React.createContext({
  lightMode: "body",
  darkMode: "bodyDark"
});
const BackgroundProvider = backgroundContext.Provider;
const useBackground = () => React.useContext(backgroundContext);
const useBackgroundLightness = (backgroundOverride) => {
  const backgroundFromContext = useBackground();
  const background = backgroundOverride || backgroundFromContext;
  const { backgroundLightness } = sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useBraidTheme();
  const lightnessMap = {
    ...backgroundLightness,
    customDark: "dark",
    customLight: "light"
  };
  return {
    lightMode: lightnessMap[background.lightMode],
    darkMode: lightnessMap[background.darkMode]
  };
};
const useColorContrast = () => {
  const background = useBackground();
  const backgroundLightness = useBackgroundLightness();
  return (map) => styles_lib_css_atoms_sprinkles_css_cjs.mapColorModeValue(
    backgroundLightness,
    (lightness, mode) => typeof map === "function" ? map(lightness, background[mode]) : map[lightness]
  );
};
const adaptiveBackgrounds = {
  body: "bodyDark",
  surface: "surfaceDark",
  brand: "neutral",
  promoteLight: "neutral",
  infoLight: "neutral",
  positiveLight: "neutral",
  cautionLight: "neutral",
  criticalLight: "neutral",
  neutralLight: "neutral",
  formAccentSoft: "neutral",
  formAccentSoftActive: "neutralActive",
  formAccentSoftHover: "neutralHover"
};
const adaptiveBoxShadow = {
  borderBrandAccent: "borderBrandAccentLight",
  borderBrandAccentLarge: "borderBrandAccentLightLarge",
  borderCaution: "borderCautionLight",
  borderCritical: "borderCriticalLight",
  borderCriticalLarge: "borderCriticalLightLarge",
  borderFormAccent: "borderFormAccentLight",
  borderFormAccentLarge: "borderFormAccentLightLarge",
  borderInfo: "borderInfoLight",
  borderNeutral: "borderNeutralInverted",
  borderNeutralLarge: "borderNeutralInvertedLarge",
  borderNeutralLight: "borderNeutral",
  borderPositive: "borderPositiveLight",
  borderPromote: "borderPromoteLight"
};
const normaliseBackground = (background) => ({
  lightMode: typeof background === "object" ? background.lightMode : background,
  darkMode: typeof background === "object" ? background.darkMode : adaptiveBackgrounds[background] || background
});
const normalisedBoxShadow = (boxShadow) => ({
  lightMode: typeof boxShadow === "object" ? boxShadow.lightMode : boxShadow,
  darkMode: typeof boxShadow === "object" ? boxShadow.darkMode : boxShadow
});
const useColoredBoxClasses = ({
  background,
  boxShadow
}) => {
  const parentLightness = useBackgroundLightness();
  const currentBackgroundContext = useBackground();
  const { backgroundLightness } = sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useBraidTheme();
  const lightnessMap = {
    ...backgroundLightness,
    customDark: "dark",
    customLight: "light"
  };
  const classList = [];
  if (boxShadow) {
    const normalisedBoxShadows = normalisedBoxShadow(boxShadow);
    const isSemantic = typeof boxShadow === "string";
    classList.push(
      styles_lib_css_atoms_atoms_cjs.atoms({
        boxShadow: {
          lightMode: isSemantic && parentLightness.lightMode === "dark" ? adaptiveBoxShadow[normalisedBoxShadows.lightMode] || normalisedBoxShadows.lightMode : normalisedBoxShadows.lightMode,
          darkMode: isSemantic && parentLightness.darkMode === "dark" ? adaptiveBoxShadow[normalisedBoxShadows.darkMode] || normalisedBoxShadows.darkMode : normalisedBoxShadows.darkMode
        }
      })
    );
  }
  if (background) {
    const { lightMode, darkMode } = normaliseBackground(background);
    classList.push(styles_lib_css_typography_css_cjs.lightModeTone[lightnessMap[lightMode]]);
    classList.push(styles_lib_css_typography_css_cjs.darkModeTone[lightnessMap[darkMode]]);
    const lightModeTone = lightMode === "neutral" ? darkMode : lightMode;
    const darkModeTone = darkMode === "neutral" ? lightMode : darkMode;
    if (lightModeTone && styles_lib_css_typography_css_cjs.lightModeNeutralOverride[lightModeTone]) {
      classList.push(styles_lib_css_typography_css_cjs.lightModeNeutralOverride[lightModeTone]);
    }
    if (darkModeTone && styles_lib_css_typography_css_cjs.darkModeNeutralOverride[darkModeTone]) {
      classList.push(styles_lib_css_typography_css_cjs.darkModeNeutralOverride[darkModeTone]);
    }
    classList.push(
      styles_lib_css_atoms_atoms_cjs.atoms({
        background: {
          lightMode: lightMode === "customDark" || lightMode === "customLight" ? void 0 : lightMode,
          darkMode: darkMode === "customDark" || darkMode === "customLight" ? void 0 : darkMode
        }
      })
    );
    return {
      backgroundContext: {
        lightMode: lightMode || currentBackgroundContext.lightMode,
        darkMode: darkMode || currentBackgroundContext.darkMode
      },
      classList: classList.join(" ")
    };
  }
  return {
    classList: classList.join(" ")
  };
};
const ColoredBox = React.forwardRef(
  ({ component, background, boxShadow, className, ...props }, ref) => {
    const { backgroundContext: backgroundContext2, classList: colorClasses } = useColoredBoxClasses(
      { background, boxShadow }
    );
    const element = React.createElement(component, {
      className: `${className}${colorClasses ? ` ${colorClasses}` : ""}`,
      ...props,
      ref
    });
    return backgroundContext2 ? /* @__PURE__ */ jsxRuntime.jsx(BackgroundProvider, { value: backgroundContext2, children: element }) : element;
  }
);
ColoredBox.displayName = "ColoredBox";
const Box = React.forwardRef(
  ({ component = "div", className, background, boxShadow, data, ...restProps }, ref) => {
    const atomProps = {};
    const nativeProps = {
      ...data ? (
        // Not validating rest props as Box supports native HTML element props
        // and we do not want to warn against using the native syntax.
        buildDataAttributes({ data, validateRestProps: false })
      ) : void 0
    };
    for (const key in restProps) {
      if (styles_lib_css_atoms_sprinkles_css_cjs.sprinkles.properties.has(key)) {
        atomProps[key] = restProps[key];
      } else {
        nativeProps[key] = restProps[key];
      }
    }
    const userClasses = clsx__default.default(className);
    if (process.env.NODE_ENV !== "production") {
      React.useEffect(() => {
        if (userClasses.includes(styles_lib_css_reset_reset_css_cjs.base)) {
          throw new Error(
            dedent__default.default`
              Reset class has been applied more than once. This is normally caused when asking for an explicit reset on the \`atoms\` function. This can be removed as Box automatically adds reset classes.

              atoms({
                reset: '...' // <-- Remove this
              })
            `
          );
        }
      }, [userClasses]);
    }
    const atomicClasses = styles_lib_css_atoms_atoms_cjs.atoms({
      reset: typeof component === "string" ? component : "div",
      ...atomProps
    });
    const combinedClasses = `${atomicClasses}${userClasses ? ` ${userClasses}` : ""}`;
    return background || boxShadow ? /* @__PURE__ */ jsxRuntime.jsx(
      ColoredBox,
      {
        component,
        background,
        boxShadow,
        className: combinedClasses,
        ref,
        ...nativeProps
      }
    ) : React.createElement(component, {
      className: combinedClasses,
      ...nativeProps,
      ref
    });
  }
);
Box.displayName = "Box";
const PublicBox = React.forwardRef(
  (props, ref) => {
    if (process.env.NODE_ENV !== "production") {
      if (typeof props.background !== "undefined" && typeof props.background !== "string") {
        throw new Error("Conditional backgrounds are not suppported");
      }
      if (typeof props.boxShadow !== "undefined" && typeof props.boxShadow !== "string") {
        throw new Error("Conditional boxShadows are not suppported");
      }
    }
    return /* @__PURE__ */ jsxRuntime.jsx(Box, { ...props, ref });
  }
);
PublicBox.displayName = "Box";
function flattenChildren(children, depth = 0, keys = []) {
  return React.Children.toArray(children).reduce(
    (acc, node, nodeIndex) => {
      if (reactIs.isFragment(node)) {
        acc.push(
          ...flattenChildren(
            node.props.children,
            depth + 1,
            keys.concat(node.key || nodeIndex)
          )
        );
      } else if (React.isValidElement(node)) {
        acc.push(
          React.cloneElement(node, {
            key: keys.concat(String(node.key)).join(".")
          })
        );
      } else if (typeof node === "string" || typeof node === "number") {
        acc.push(node);
      }
      return acc;
    },
    []
  );
}
const Divider = ({ weight = "regular" }) => {
  const lightness = useBackgroundLightness();
  return /* @__PURE__ */ jsxRuntime.jsx(Box, { component: "span", display: "block", position: "relative", children: /* @__PURE__ */ jsxRuntime.jsx(
    Box,
    {
      component: "span",
      position: "absolute",
      width: "full",
      className: [
        styles_lib_components_Divider_Divider_css_cjs.base,
        weight === "strong" ? styles_lib_components_Divider_Divider_css_cjs.strong : styles_lib_components_Divider_Divider_css_cjs.regular,
        styles_lib_components_Divider_Divider_css_cjs.lightModeWeight[lightness.lightMode],
        styles_lib_components_Divider_Divider_css_cjs.darkModeWeight[lightness.darkMode]
      ]
    }
  ) });
};
const HeadingContext = React.createContext(false);
const resolveResponsiveRangeProps = (props) => {
  const { above, below } = props;
  if (!above && !below) {
    return [false, false, false, false];
  }
  const startIndex = above ? styles_lib_css_breakpoints_cjs.breakpointNames.indexOf(above) + 1 : 0;
  const endIndex = below ? styles_lib_css_breakpoints_cjs.breakpointNames.indexOf(below) - 1 : styles_lib_css_breakpoints_cjs.breakpointNames.length - 1;
  const range = styles_lib_css_breakpoints_cjs.breakpointNames.slice(startIndex, endIndex + 1);
  const includeMobile = range.indexOf("mobile") >= 0;
  const includeTablet = range.indexOf("tablet") >= 0;
  const includeDesktop = range.indexOf("desktop") >= 0;
  const includeWide = range.indexOf("wide") >= 0;
  return [includeMobile, includeTablet, includeDesktop, includeWide];
};
const optimizeResponsiveArray = (value) => {
  let lastValue;
  const values = value.map((v) => {
    if (v !== lastValue && v !== null) {
      lastValue = v;
      return v;
    }
    return null;
  });
  return [
    values[0] ?? null,
    values[1] ?? null,
    values[2] ?? null,
    values[3] ?? null
  ];
};
const Hidden = ({
  children,
  component,
  above,
  below,
  screen,
  print,
  inline: inlineProp,
  data,
  ...restProps
}) => {
  if (process.env.NODE_ENV === "development" && screen) {
    console.warn(
      `You used the "screen" prop on Hidden, but this probably doesn't do what you expect. If you're trying to provide content to screen readers without rendering it to the screen, use the <HiddenVisually> component instead. The "screen" prop is likely to be deprecated in a future release.`
    );
  }
  const inText = Boolean(React.useContext(TextContext));
  const inHeading = Boolean(React.useContext(HeadingContext));
  const hiddenOnScreen = Boolean(screen);
  const hiddenOnPrint = Boolean(print);
  const [hiddenOnMobile, hiddenOnTablet, hiddenOnDesktop, hiddenOnWide] = resolveResponsiveRangeProps({ above, below });
  const inline = inlineProp ?? (inText || inHeading);
  const display = inline ? "inline" : "block";
  return /* @__PURE__ */ jsxRuntime.jsx(
    Box,
    {
      display: hiddenOnScreen ? "none" : optimizeResponsiveArray([
        hiddenOnMobile ? "none" : display,
        hiddenOnTablet ? "none" : display,
        hiddenOnDesktop ? "none" : display,
        hiddenOnWide ? "none" : display
      ]),
      className: hiddenOnPrint ? styles_lib_components_Hidden_Hidden_css_cjs.hiddenOnPrint : void 0,
      component: component || (inline ? "span" : "div"),
      ...buildDataAttributes({ data, validateRestProps: restProps }),
      children
    }
  );
};
const alignToFlexAlignLookup = {
  left: "flexStart",
  center: "center",
  right: "flexEnd"
};
const alignToFlexAlign = (align) => align ? styles_lib_css_atoms_sprinkles_css_cjs.mapResponsiveValue(align, (value) => alignToFlexAlignLookup[value]) : void 0;
const alignYToFlexAlignLookup = {
  top: "flexStart",
  center: "center",
  bottom: "flexEnd"
};
const alignYToFlexAlign = (alignY) => alignY ? styles_lib_css_atoms_sprinkles_css_cjs.mapResponsiveValue(alignY, (value) => alignYToFlexAlignLookup[value]) : void 0;
const resolveResponsiveProp = (value, mobileAtoms, tabletAtoms, desktopAtoms, wideAtoms) => {
  if (typeof value === "string" || typeof value === "number") {
    return mobileAtoms[value];
  }
  const normalized = styles_lib_css_atoms_sprinkles_css_cjs.normalizeResponsiveValue(value);
  const [mobile, tablet, desktop, wide] = optimizeResponsiveArray([
    normalized.mobile ?? null,
    normalized.tablet ?? null,
    normalized.desktop ?? null,
    normalized.wide ?? null
  ]);
  const mobileAtom = mobileAtoms[mobile];
  const tabletAtom = tabletAtoms[tablet];
  const desktopAtom = desktopAtoms[desktop];
  const wideAtom = wideAtoms[wide];
  return clsx__default.default(mobileAtom, tabletAtom, desktopAtom, wideAtom);
};
const directionStyles = {
  top: styles_lib_css_negativeMargin_negativeMargin_css_cjs.top,
  right: styles_lib_css_negativeMargin_negativeMargin_css_cjs.right,
  bottom: styles_lib_css_negativeMargin_negativeMargin_css_cjs.bottom,
  left: styles_lib_css_negativeMargin_negativeMargin_css_cjs.left
};
const negativeMargin = (direction, space) => space ? clsx__default.default([
  direction === "top" || direction === "bottom" ? styles_lib_css_negativeMargin_negativeMargin_css_cjs.preventCollapsePseudo[direction] : void 0,
  resolveResponsiveProp(
    space,
    directionStyles[direction].mobile,
    directionStyles[direction].tablet,
    directionStyles[direction].desktop,
    directionStyles[direction].wide
  )
]) : null;
const alignToDisplay = {
  left: "block",
  center: "flex",
  right: "flex"
};
const validStackComponents = ["div", "span", "ol", "ul"];
const useStackItem = ({ align, space, component }) => ({
  paddingTop: space,
  display: component === "span" ? "block" : void 0,
  // If we're aligned left across all screen sizes,
  // there's actually no alignment work to do.
  ...align === "left" ? null : {
    display: styles_lib_css_atoms_sprinkles_css_cjs.mapResponsiveValue(align, (value) => alignToDisplay[value]),
    flexDirection: "column",
    alignItems: alignToFlexAlign(align)
  }
});
const extractHiddenPropsFromChild = (child) => child && typeof child === "object" && "type" in child && child.type === Hidden ? child.props : null;
const resolveHiddenProps = ({ screen, above, below }) => screen ? [true, true, true, true] : resolveResponsiveRangeProps({
  above,
  below
});
const calculateHiddenStackItemProps = (stackItemProps, [hiddenOnMobile, hiddenOnTablet, hiddenOnDesktop, hiddenOnWide]) => {
  const normalizedValue = styles_lib_css_atoms_sprinkles_css_cjs.normalizeResponsiveValue(
    stackItemProps.display !== void 0 ? stackItemProps.display : "block"
  );
  const {
    mobile: displayMobile = "block",
    tablet: displayTablet = displayMobile,
    desktop: displayDesktop = displayTablet,
    wide: displayWide = displayDesktop
  } = normalizedValue;
  return {
    ...stackItemProps,
    display: optimizeResponsiveArray([
      hiddenOnMobile ? "none" : displayMobile,
      hiddenOnTablet ? "none" : displayTablet,
      hiddenOnDesktop ? "none" : displayDesktop,
      hiddenOnWide ? "none" : displayWide
    ])
  };
};
const Stack = ({
  component = "div",
  children,
  space = "none",
  align = "left",
  dividers = false,
  data,
  ...restProps
}) => {
  assert__default.default(
    validStackComponents.includes(component),
    `Invalid Stack component: '${component}'. Should be one of [${validStackComponents.map((c) => `'${c}'`).join(", ")}]`
  );
  const stackItemProps = useStackItem({ space, align, component });
  const stackItems = flattenChildren(children);
  const isList = component === "ol" || component === "ul";
  const stackItemComponent = isList ? "li" : component;
  let firstItemOnMobile = null;
  let firstItemOnTablet = null;
  let firstItemOnDesktop = null;
  let firstItemOnWide = null;
  return /* @__PURE__ */ jsxRuntime.jsx(
    Box,
    {
      component,
      display: component === "span" ? "block" : void 0,
      className: negativeMargin("top", space),
      ...buildDataAttributes({ data, validateRestProps: restProps }),
      children: React.Children.map(stackItems, (child, index) => {
        assert__default.default(
          !(typeof child === "object" && child.type === Hidden && child.props.inline !== void 0),
          'The "inline" prop is invalid on Hidden elements within a Stack'
        );
        const hiddenProps = extractHiddenPropsFromChild(child);
        const hidden = hiddenProps ? resolveHiddenProps(hiddenProps) : [false, false, false, false];
        const [hiddenOnMobile, hiddenOnTablet, hiddenOnDesktop, hiddenOnWide] = hidden;
        const responsivelyHidden = hiddenOnMobile || hiddenOnTablet || hiddenOnDesktop || hiddenOnWide;
        if (firstItemOnMobile === null && !hiddenOnMobile) {
          firstItemOnMobile = index;
        }
        if (firstItemOnTablet === null && !hiddenOnTablet) {
          firstItemOnTablet = index;
        }
        if (firstItemOnDesktop === null && !hiddenOnDesktop) {
          firstItemOnDesktop = index;
        }
        if (firstItemOnWide === null && !hiddenOnWide) {
          firstItemOnWide = index;
        }
        return /* @__PURE__ */ jsxRuntime.jsxs(
          Box,
          {
            component: stackItemComponent,
            className: [
              hiddenProps && hiddenProps.print ? styles_lib_components_Hidden_Hidden_css_cjs.hiddenOnPrint : null
            ],
            ...responsivelyHidden ? calculateHiddenStackItemProps(stackItemProps, hidden) : stackItemProps,
            children: [
              dividers && index > 0 ? /* @__PURE__ */ jsxRuntime.jsx(
                Box,
                {
                  component: "span",
                  width: "full",
                  paddingBottom: space,
                  display: optimizeResponsiveArray([
                    index === firstItemOnMobile ? "none" : "block",
                    index === firstItemOnTablet ? "none" : "block",
                    index === firstItemOnDesktop ? "none" : "block",
                    index === firstItemOnWide ? "none" : "block"
                  ]),
                  children: typeof dividers === "string" ? /* @__PURE__ */ jsxRuntime.jsx(Divider, { weight: dividers }) : /* @__PURE__ */ jsxRuntime.jsx(Divider, {})
                }
              ) : null,
              hiddenProps ? hiddenProps.children : child
            ]
          }
        );
      })
    }
  );
};
const DefaultTextPropsContext = React.createContext({
  tone: void 0,
  weight: void 0,
  size: void 0
});
const DefaultTextPropsProvider = ({
  size,
  weight,
  tone,
  children
}) => {
  const defaultTextProps = React.useMemo(
    () => ({
      size,
      weight,
      tone
    }),
    [size, weight, tone]
  );
  return /* @__PURE__ */ jsxRuntime.jsx(DefaultTextPropsContext.Provider, { value: defaultTextProps, children });
};
const useDefaultTextProps = ({
  size: sizeProp,
  weight: weightProp,
  tone: toneProp
}) => {
  const { size, weight, tone } = React.useContext(DefaultTextPropsContext);
  return {
    size: sizeProp ?? size ?? "standard",
    weight: weightProp ?? weight ?? "regular",
    tone: toneProp ?? tone ?? "neutral"
  };
};
function textStyles({
  weight = "regular",
  size = "standard",
  tone = "neutral",
  baseline
}) {
  return [
    styles_lib_css_typography_css_cjs.fontFamily,
    styles_lib_css_typography_css_cjs.fontWeight[weight],
    styles_lib_css_typography_css_cjs.tone[tone],
    (baseline ? styles_lib_css_typography_css_cjs.textSizeTrimmed : styles_lib_css_typography_css_cjs.textSizeUntrimmed)[size]
  ];
}
const MaxLines = ({
  children,
  lines
}) => /* @__PURE__ */ jsxRuntime.jsx(
  Box,
  {
    component: "span",
    className: [styles_lib_components_private_MaxLines_MaxLines_css_cjs.base, lines > 1 ? styles_lib_components_private_MaxLines_MaxLines_css_cjs.multiLine : void 0],
    style: lines > 1 ? dynamic.assignInlineVars({ [styles_lib_components_private_MaxLines_MaxLines_css_cjs.lineLimit]: String(lines) }) : void 0,
    children
  }
);
const Typography = ({
  id,
  component = "span",
  className,
  align,
  truncate = false,
  maxLines,
  icon,
  data,
  children,
  ...restProps
}) => {
  const lines = truncate ? 1 : maxLines;
  const contents = typeof lines === "number" ? /* @__PURE__ */ jsxRuntime.jsx(MaxLines, { lines, children }) : children;
  if (process.env.NODE_ENV !== "production") {
    if (truncate) {
      console.warn(
        dedent__default.default`
          The "truncate" prop has been deprecated and will be removed in a future version. Use "maxLines" instead.
             <Text
            %c-   truncate
            %c+   maxLines={1}
             %c/>
        `,
        "color: red",
        "color: green",
        "color: inherit"
      );
    }
  }
  return /* @__PURE__ */ jsxRuntime.jsx(
    Box,
    {
      id,
      display: "block",
      component,
      textAlign: align,
      className,
      ...buildDataAttributes({ data, validateRestProps: restProps }),
      children: icon ? /* @__PURE__ */ jsxRuntime.jsxs(
        Box,
        {
          component: "span",
          display: "flex",
          justifyContent: alignToFlexAlign(align),
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(
              Box,
              {
                component: "span",
                display: "block",
                paddingRight: "xsmall",
                flexGrow: 0,
                flexShrink: 0,
                minWidth: 0,
                textAlign: align,
                children: icon
              }
            ),
            /* @__PURE__ */ jsxRuntime.jsx(Box, { component: "span", display: "block", minWidth: 0, children: contents })
          ]
        }
      ) : contents
    }
  );
};
const Text = ({
  size: sizeProp,
  tone: toneProp,
  weight: weightProp,
  baseline = true,
  ...typographyProps
}) => {
  assert__default.default(
    !React.useContext(TextContext),
    "Text components should not be nested within each other"
  );
  const { size, weight, tone } = useDefaultTextProps({
    size: sizeProp,
    weight: weightProp,
    tone: toneProp
  });
  const textStylingProps = React.useMemo(
    () => ({
      tone,
      size,
      weight,
      baseline
    }),
    [tone, size, weight, baseline]
  );
  return /* @__PURE__ */ jsxRuntime.jsx(TextContext.Provider, { value: textStylingProps, children: /* @__PURE__ */ jsxRuntime.jsx(
    Typography,
    {
      ...typographyProps,
      className: textStyles(textStylingProps)
    }
  ) });
};
function invertAlignment(alignment) {
  if (alignment === "flexStart") {
    return "flexEnd";
  }
  if (alignment === "flexEnd") {
    return "flexStart";
  }
  return alignment;
}
function resolveCollapsibleAlignmentProps({
  align,
  alignY,
  collapseBelow,
  reverse
}) {
  const [collapseMobile, collapseTablet, collapseDesktop] = resolveResponsiveRangeProps({
    below: collapseBelow
  });
  const rowReverseTablet = collapseMobile && reverse;
  const rowReverseDesktop = (collapseMobile || collapseTablet) && reverse;
  const rowReverseWide = (collapseMobile || collapseTablet || collapseDesktop) && reverse;
  const normalizedAlign = styles_lib_css_atoms_sprinkles_css_cjs.normalizeResponsiveValue(
    alignToFlexAlign(align) || "flexStart"
  );
  const {
    mobile: justifyContentMobile = "flexStart",
    tablet: justifyContentTablet = justifyContentMobile,
    desktop: justifyContentDesktop = justifyContentTablet,
    wide: justifyContentWide = justifyContentDesktop
  } = normalizedAlign;
  return {
    collapseMobile,
    collapseTablet,
    collapseDesktop,
    orderChildren: (children) => {
      const childrenArray = React.Children.toArray(children);
      return !collapseMobile && !collapseTablet && reverse ? childrenArray.reverse() : childrenArray;
    },
    collapsibleAlignmentProps: {
      display: optimizeResponsiveArray([
        collapseMobile ? "block" : "flex",
        collapseTablet ? "block" : "flex",
        collapseDesktop ? "block" : "flex",
        "flex"
      ]),
      flexDirection: optimizeResponsiveArray([
        collapseMobile ? "column" : "row",
        // eslint-disable-next-line no-nested-ternary
        collapseTablet ? "column" : rowReverseTablet ? "rowReverse" : "row",
        // eslint-disable-next-line no-nested-ternary
        collapseDesktop ? "column" : rowReverseDesktop ? "rowReverse" : "row",
        rowReverseWide ? "rowReverse" : "row"
      ]),
      justifyContent: align ? optimizeResponsiveArray([
        justifyContentMobile,
        rowReverseTablet ? invertAlignment(justifyContentTablet) : justifyContentTablet,
        rowReverseDesktop ? invertAlignment(justifyContentDesktop) : justifyContentDesktop,
        rowReverseWide ? invertAlignment(justifyContentWide) : justifyContentWide
      ]) : void 0,
      alignItems: alignY ? alignYToFlexAlign(alignY) : void 0
    },
    collapsibleAlignmentChildProps: {
      display: optimizeResponsiveArray([
        collapseMobile && justifyContentMobile !== "flexStart" ? "flex" : "block",
        collapseTablet && justifyContentTablet !== "flexStart" ? "flex" : "block",
        collapseDesktop && justifyContentDesktop !== "flexStart" ? "flex" : "block",
        "block"
      ]),
      justifyContent: optimizeResponsiveArray([
        justifyContentMobile,
        justifyContentTablet,
        justifyContentDesktop,
        justifyContentWide
      ])
    }
  };
}
const validColumnsComponents = ["div", "span"];
const ColumnsContext = React.createContext({
  collapseMobile: false,
  collapseTablet: false,
  collapseDesktop: false,
  mobileSpace: "none",
  tabletSpace: "none",
  desktopSpace: "none",
  wideSpace: "none",
  collapsibleAlignmentChildProps: null,
  component: "div"
});
const Columns = ({
  children,
  collapseBelow,
  reverse = false,
  space = "none",
  align,
  alignY,
  component = "div",
  data,
  ...restProps
}) => {
  assert__default.default(
    validColumnsComponents.includes(component),
    `Invalid Columns component: '${component}'. Should be one of [${validColumnsComponents.map((c) => `'${c}'`).join(", ")}]`
  );
  const normalizedSpace = styles_lib_css_atoms_sprinkles_css_cjs.normalizeResponsiveValue(space);
  const {
    mobile: mobileSpace = "none",
    tablet: tabletSpace = mobileSpace,
    desktop: desktopSpace = tabletSpace,
    wide: wideSpace = desktopSpace
  } = normalizedSpace;
  const {
    collapsibleAlignmentProps,
    collapsibleAlignmentChildProps,
    collapseMobile,
    collapseTablet,
    collapseDesktop,
    orderChildren
  } = resolveCollapsibleAlignmentProps({
    collapseBelow,
    align,
    alignY,
    reverse
  });
  return /* @__PURE__ */ jsxRuntime.jsx(
    Box,
    {
      component,
      ...collapsibleAlignmentProps,
      className: negativeMargin("left", {
        mobile: collapseMobile ? "none" : mobileSpace,
        tablet: collapseTablet ? "none" : tabletSpace,
        desktop: collapseDesktop ? "none" : desktopSpace,
        wide: wideSpace
      }),
      ...buildDataAttributes({ data, validateRestProps: restProps }),
      children: /* @__PURE__ */ jsxRuntime.jsx(
        ColumnsContext.Provider,
        {
          value: {
            collapseMobile,
            collapseTablet,
            collapseDesktop,
            mobileSpace,
            tabletSpace,
            desktopSpace,
            wideSpace,
            collapsibleAlignmentChildProps,
            component
          },
          children: orderChildren(children)
        }
      )
    }
  );
};
const Column = ({
  children,
  data,
  width,
  ...restProps
}) => {
  const {
    collapseMobile,
    collapseTablet,
    collapseDesktop,
    mobileSpace,
    tabletSpace,
    desktopSpace,
    wideSpace,
    collapsibleAlignmentChildProps,
    component
  } = React.useContext(ColumnsContext);
  return /* @__PURE__ */ jsxRuntime.jsx(
    Box,
    {
      component,
      display: component === "span" ? "block" : void 0,
      minWidth: 0,
      width: width !== "content" ? "full" : void 0,
      flexShrink: width === "content" ? 0 : void 0,
      flexGrow: 1,
      className: [
        styles_lib_components_Column_Column_css_cjs.column,
        width !== "content" ? styles_lib_components_Column_Column_css_cjs.width[width] : null
      ],
      ...buildDataAttributes({ data, validateRestProps: restProps }),
      children: /* @__PURE__ */ jsxRuntime.jsx(
        Box,
        {
          component,
          paddingLeft: optimizeResponsiveArray([
            collapseMobile ? "none" : mobileSpace,
            collapseTablet ? "none" : tabletSpace,
            collapseDesktop ? "none" : desktopSpace,
            wideSpace
          ]),
          paddingTop: collapseMobile || collapseTablet || collapseDesktop ? optimizeResponsiveArray([
            collapseMobile ? mobileSpace : "none",
            collapseTablet ? tabletSpace : "none",
            collapseDesktop ? desktopSpace : "none",
            "none"
          ]) : void 0,
          height: "full",
          ...collapsibleAlignmentChildProps,
          className: styles_lib_components_Column_Column_css_cjs.columnContent,
          children
        }
      )
    }
  );
};
const validInlineComponents = ["div", "span", "ol", "ul"];
const Inline = ({
  space = "none",
  align,
  alignY,
  collapseBelow,
  reverse,
  component = "div",
  data,
  children,
  ...restProps
}) => {
  assert__default.default(
    validInlineComponents.includes(component),
    `Invalid Inline component: '${component}'. Should be one of [${validInlineComponents.map((c) => `'${c}'`).join(", ")}]`
  );
  const isList = component === "ol" || component === "ul";
  const inlineItemComponent = isList ? "li" : component;
  const {
    collapsibleAlignmentProps,
    collapsibleAlignmentChildProps,
    orderChildren
  } = resolveCollapsibleAlignmentProps({
    align,
    alignY,
    collapseBelow,
    reverse
  });
  return /* @__PURE__ */ jsxRuntime.jsx(
    Box,
    {
      component: component === "span" ? component : void 0,
      display: component === "span" ? "block" : void 0,
      className: negativeMargin("top", space),
      ...buildDataAttributes({ data, validateRestProps: restProps }),
      children: /* @__PURE__ */ jsxRuntime.jsx(
        Box,
        {
          component,
          className: negativeMargin("left", space),
          flexWrap: "wrap",
          ...collapsibleAlignmentProps,
          children: React.Children.map(
            orderChildren(flattenChildren(children)),
            (child) => child !== null && child !== void 0 ? /* @__PURE__ */ jsxRuntime.jsx(
              Box,
              {
                component: inlineItemComponent,
                minWidth: 0,
                marginLeft: space,
                marginTop: space,
                ...collapsibleAlignmentChildProps,
                children: child
              }
            ) : null
          )
        }
      )
    }
  );
};
const iconInlineSize = ({
  alignY = "uppercase",
  verticalCorrection = "none"
} = {}) => clsx__default.default(
  styles_lib_css_atoms_atoms_cjs.atoms({
    display: "inlineBlock",
    position: "relative"
  }),
  styles_lib_hooks_useIcon_icon_css_cjs.size,
  styles_lib_hooks_useIcon_icon_css_cjs.inlineCrop,
  styles_lib_hooks_useIcon_icon_css_cjs.inline,
  styles_lib_hooks_useIcon_icon_css_cjs.alignY[alignY][verticalCorrection]
);
const IconRenderer = ({ children }) => {
  const textContext = React.useContext(TextContext);
  const headingContext = React.useContext(HeadingContext);
  assert__default.default(
    Boolean(textContext || headingContext),
    `IconRenderer must be inside either a \`Text\` or \`Heading\` component.`
  );
  return children({ className: iconInlineSize() });
};
const iconSize = ({
  size = "standard",
  crop = false
} = {}) => clsx__default.default(styles_lib_hooks_useIcon_icon_css_cjs.size, styles_lib_css_typography_css_cjs.textSizeUntrimmed[size], {
  [styles_lib_hooks_useIcon_icon_css_cjs.cropToTextSize]: crop
});
const iconContainerSize = (size = "standard") => clsx__default.default(styles_lib_hooks_useIcon_icon_css_cjs.blockWidths[size], styles_lib_css_lineHeightContainer_css_cjs.lineHeightContainer[size]);
const defaultVerticalCorrection = {
  uppercase: "none",
  lowercase: "none"
};
const useIcon = ({ size, tone, alignY, data, title, titleId, ...restProps }, { verticalCorrection = defaultVerticalCorrection } = {}) => {
  const textContext = React.useContext(TextContext);
  const headingContext = React.useContext(HeadingContext);
  const resolvedTone = styles_lib_css_typography_css_cjs.tone[tone || (textContext == null ? void 0 : textContext.tone) || "neutral"];
  const toneClass = tone || !headingContext && !(textContext == null ? void 0 : textContext.tone) ? resolvedTone : void 0;
  const isInline = Boolean(textContext || headingContext);
  const a11yProps = title ? { title, titleId, role: "img" } : { "aria-hidden": true };
  assert__default.default(
    !(size && isInline),
    `Specifying a custom \`size\` for an \`Icon\` inside the context of a \`<${textContext ? "Text" : "Heading"}>\` component is invalid. See the documentation for correct usage: https://seek-oss.github.io/braid-design-system/components/${textContext ? "Text" : "Heading"}`
  );
  assert__default.default(
    !(alignY && !isInline),
    `Specifying \`alignY\` for an \`Icon\` outside of a text component is invalid.`
  );
  if (size === "fill") {
    return {
      width: "full",
      height: "full",
      display: "block",
      className: toneClass,
      ...buildDataAttributes({ data, validateRestProps: restProps }),
      ...a11yProps
    };
  }
  return {
    className: [
      toneClass,
      isInline ? [
        iconInlineSize({
          alignY: alignY || "uppercase",
          verticalCorrection: verticalCorrection[alignY || "uppercase"]
        })
      ] : [
        styles_lib_css_atoms_atoms_cjs.atoms({
          display: "block"
        }),
        iconContainerSize(size)
      ]
    ],
    ...buildDataAttributes({ data, validateRestProps: restProps }),
    ...a11yProps
  };
};
const IconClearSvg = ({ title, titleId, ...props }) => /* @__PURE__ */ jsxRuntime.jsxs(
  "svg",
  {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 24 24",
    xmlSpace: "preserve",
    focusable: "false",
    fill: "currentColor",
    width: 16,
    height: 16,
    "aria-labelledby": titleId,
    ...props,
    children: [
      title ? /* @__PURE__ */ jsxRuntime.jsx("title", { id: titleId, children: title }) : null,
      /* @__PURE__ */ jsxRuntime.jsx("path", { d: "m13.4 12 5.3-5.3c.4-.4.4-1 0-1.4s-1-.4-1.4 0L12 10.6 6.7 5.3c-.4-.4-1-.4-1.4 0s-.4 1 0 1.4l5.3 5.3-5.3 5.3c-.4.4-.4 1 0 1.4.2.2.4.3.7.3s.5-.1.7-.3l5.3-5.3 5.3 5.3c.2.2.5.3.7.3s.5-.1.7-.3c.4-.4.4-1 0-1.4L13.4 12z" })
    ]
  }
);
const IconClear = (props) => {
  const iconProps = useIcon(props);
  return /* @__PURE__ */ jsxRuntime.jsx(Box, { component: IconClearSvg, ...iconProps });
};
const IconCriticalSvg = ({ title, titleId, ...props }) => /* @__PURE__ */ jsxRuntime.jsxs(
  "svg",
  {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 24 24",
    xmlSpace: "preserve",
    focusable: "false",
    fill: "currentColor",
    width: 16,
    height: 16,
    "aria-labelledby": titleId,
    ...props,
    children: [
      title ? /* @__PURE__ */ jsxRuntime.jsx("title", { id: titleId, children: title }) : null,
      /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M12 1C5.9 1 1 5.9 1 12s4.9 11 11 11 11-4.9 11-11S18.1 1 12 1zm0 20c-5 0-9-4-9-9s4-9 9-9 9 4 9 9-4 9-9 9z" }),
      /* @__PURE__ */ jsxRuntime.jsx("circle", { cx: 12, cy: 17, r: 1 }),
      /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M12 14c.6 0 1-.4 1-1V8c0-.6-.4-1-1-1s-1 .4-1 1v5c0 .6.4 1 1 1z" })
    ]
  }
);
const IconCritical = (props) => {
  const iconProps = useIcon(props);
  return /* @__PURE__ */ jsxRuntime.jsx(Box, { component: IconCriticalSvg, ...iconProps });
};
const IconPositiveSvg = ({ title, titleId, ...props }) => /* @__PURE__ */ jsxRuntime.jsxs(
  "svg",
  {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 24 24",
    xmlSpace: "preserve",
    focusable: "false",
    fill: "currentColor",
    width: 16,
    height: 16,
    "aria-labelledby": titleId,
    ...props,
    children: [
      title ? /* @__PURE__ */ jsxRuntime.jsx("title", { id: titleId, children: title }) : null,
      /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M12 1C5.9 1 1 5.9 1 12s4.9 11 11 11 11-4.9 11-11S18.1 1 12 1zm0 20c-5 0-9-4-9-9s4-9 9-9 9 4 9 9-4 9-9 9z" }),
      /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M15.3 9.3 11 13.6l-1.3-1.3c-.4-.4-1-.4-1.4 0s-.4 1 0 1.4l2 2c.2.2.5.3.7.3s.5-.1.7-.3l5-5c.4-.4.4-1 0-1.4s-1-.4-1.4 0z" })
    ]
  }
);
const IconPositive = (props) => {
  const iconProps = useIcon(props);
  return /* @__PURE__ */ jsxRuntime.jsx(Box, { component: IconPositiveSvg, ...iconProps });
};
function virtualTouchable({ xAxis } = { xAxis: true }) {
  return !xAxis ? [styles_lib_components_private_touchable_virtualTouchable_css_cjs.virtualTouchable, styles_lib_components_private_touchable_virtualTouchable_css_cjs.yAxisOnly] : styles_lib_components_private_touchable_virtualTouchable_css_cjs.virtualTouchable;
}
const Overlay = ({
  component,
  zIndex,
  background,
  borderRadius: borderRadius2,
  boxShadow,
  transition,
  visible = false,
  onlyVisibleForKeyboardNavigation = false,
  className,
  children
}) => /* @__PURE__ */ jsxRuntime.jsx(
  Box,
  {
    component,
    position: "absolute",
    zIndex,
    pointerEvents: "none",
    background,
    borderRadius: borderRadius2,
    boxShadow,
    transition,
    inset: 0,
    opacity: !visible ? 0 : void 0,
    className: [
      onlyVisibleForKeyboardNavigation ? styles_lib_components_private_hideFocusRings_hideFocusRings_css_cjs.hideFocusRingsClassName : null,
      className
    ],
    children
  }
);
const ActionsContext = React.createContext(null);
const boxShadowForVariant = {
  default: "borderField",
  disabled: "borderNeutralLight",
  focus: "outlineFocus",
  formAccent: "borderFormAccent",
  critical: "borderCritical"
};
const FieldOverlay = ({ variant, ...restProps }) => /* @__PURE__ */ jsxRuntime.jsx(
  Overlay,
  {
    component: "span",
    borderRadius: "standard",
    boxShadow: boxShadowForVariant[variant],
    transition: "fast",
    ...restProps
  }
);
const AvoidWidowIcon = ({
  children,
  iconPosition,
  space = "xxsmall",
  className
}) => /* @__PURE__ */ jsxRuntime.jsxs(
  Box,
  {
    component: "span",
    paddingRight: iconPosition === "leading" ? space : void 0,
    paddingLeft: iconPosition === "trailing" ? space : void 0,
    className: [styles_lib_components_private_AvoidWidowIcon_AvoidWidowIcon_css_cjs.nowrap, className],
    "aria-hidden": true,
    children: [
      iconPosition === "leading" ? children : void 0,
      "⁠",
      iconPosition === "trailing" ? children : void 0
    ]
  }
);
const validBleedComponents = ["div", "span"];
const Bleed = ({
  space,
  horizontal,
  vertical,
  top,
  bottom,
  left,
  right,
  children,
  component = "div",
  data,
  ...restProps
}) => /* @__PURE__ */ jsxRuntime.jsx(
  Box,
  {
    component,
    display: component === "span" ? "block" : void 0,
    className: [
      negativeMargin("top", top || vertical || space),
      negativeMargin("bottom", bottom || vertical || space),
      negativeMargin("left", left || horizontal || space),
      negativeMargin("right", right || horizontal || space)
    ],
    ...buildDataAttributes({ data, validateRestProps: restProps }),
    children: /* @__PURE__ */ jsxRuntime.jsx(
      Box,
      {
        component,
        display: component === "span" ? "block" : void 0,
        position: "relative",
        children
      }
    )
  }
);
const buttonVariants = [
  "solid",
  "ghost",
  "soft",
  "transparent"
];
const buttonTones = [
  "formAccent",
  "brandAccent",
  "critical",
  "neutral"
];
const variants = {
  solid: {
    formAccent: {
      textTone: void 0,
      background: "formAccent",
      backgroundHover: "formAccentHover",
      backgroundActive: "formAccentActive",
      boxShadow: void 0
    },
    brandAccent: {
      textTone: void 0,
      background: "brandAccent",
      backgroundHover: "brandAccentHover",
      backgroundActive: "brandAccentActive",
      boxShadow: void 0
    },
    critical: {
      textTone: void 0,
      background: "critical",
      backgroundHover: "criticalHover",
      backgroundActive: "criticalActive",
      boxShadow: void 0
    },
    neutral: {
      textTone: void 0,
      background: { light: "neutral", dark: "neutralSoft" },
      backgroundHover: { light: "neutralSoftHover", dark: "neutralHover" },
      backgroundActive: { light: "neutralSoftActive", dark: "neutralActive" },
      boxShadow: void 0
    }
  },
  soft: {
    formAccent: {
      textTone: "formAccent",
      background: { light: "formAccentSoft", dark: "customDark" },
      backgroundHover: "formAccentSoftHover",
      backgroundActive: "formAccentSoftActive",
      boxShadow: void 0
    },
    brandAccent: {
      textTone: "brandAccent",
      background: { light: "brandAccentSoft", dark: "customDark" },
      backgroundHover: "brandAccentSoftHover",
      backgroundActive: "brandAccentSoftActive",
      boxShadow: void 0
    },
    critical: {
      textTone: "critical",
      background: { light: "criticalSoft", dark: "customDark" },
      backgroundHover: "criticalSoftHover",
      backgroundActive: "criticalSoftActive",
      boxShadow: void 0
    },
    neutral: {
      textTone: "neutral",
      background: { light: "neutralSoft", dark: "customDark" },
      backgroundHover: "neutralSoftHover",
      backgroundActive: "neutralSoftActive",
      boxShadow: void 0
    }
  },
  transparent: {
    formAccent: {
      textTone: "formAccent",
      background: void 0,
      backgroundHover: "formAccentSoftHover",
      backgroundActive: "formAccentSoftActive",
      boxShadow: void 0
    },
    brandAccent: {
      textTone: "brandAccent",
      background: void 0,
      backgroundHover: "brandAccentSoftHover",
      backgroundActive: "brandAccentSoftActive",
      boxShadow: void 0
    },
    critical: {
      textTone: "critical",
      background: void 0,
      backgroundHover: "criticalSoftHover",
      backgroundActive: "criticalSoftActive",
      boxShadow: void 0
    },
    neutral: {
      textTone: "neutral",
      background: void 0,
      backgroundHover: "neutralSoftHover",
      backgroundActive: "neutralSoftActive",
      boxShadow: void 0
    }
  },
  ghost: {
    formAccent: {
      textTone: "formAccent",
      background: void 0,
      backgroundHover: "formAccentSoftHover",
      backgroundActive: "formAccentSoftActive",
      boxShadow: "borderFormAccent"
    },
    brandAccent: {
      textTone: "brandAccent",
      background: void 0,
      backgroundHover: "brandAccentSoftHover",
      backgroundActive: "brandAccentSoftActive",
      boxShadow: "borderBrandAccent"
    },
    critical: {
      textTone: "critical",
      background: void 0,
      backgroundHover: "criticalSoftHover",
      backgroundActive: "criticalSoftActive",
      boxShadow: "borderCritical"
    },
    neutral: {
      textTone: "neutral",
      background: void 0,
      backgroundHover: "neutralSoftHover",
      backgroundActive: "neutralSoftActive",
      boxShadow: "borderNeutral"
    }
  }
};
const ButtonLoader = () => /* @__PURE__ */ jsxRuntime.jsxs(Box, { "aria-hidden": true, component: "span", display: "inlineBlock", children: [
  /* @__PURE__ */ jsxRuntime.jsx(Box, { component: "span", className: styles_lib_components_Button_Button_css_cjs.loadingDot, children: "." }),
  /* @__PURE__ */ jsxRuntime.jsx(Box, { component: "span", className: styles_lib_components_Button_Button_css_cjs.loadingDot, children: "." }),
  /* @__PURE__ */ jsxRuntime.jsx(Box, { component: "span", className: styles_lib_components_Button_Button_css_cjs.loadingDot, children: "." })
] });
const transparentPaddingX = "small";
const buttonRadius = "standard";
const resolveToneAndVariant = ({
  variant: variantProp,
  tone: toneProp,
  legacy
}) => {
  if (legacy) {
    return {
      variant: variantProp ?? "solid",
      tone: toneProp ?? "formAccent"
    };
  }
  const fallbackVariant = toneProp ? "solid" : "ghost";
  return {
    variant: variantProp ?? fallbackVariant,
    tone: toneProp ?? "neutral"
  };
};
const ButtonOverlays = ({
  variant: variantProp,
  tone: toneProp,
  keyboardFocusable = true,
  forceActive = false,
  radius = buttonRadius
}) => {
  const { variant, tone } = resolveToneAndVariant({
    variant: variantProp,
    tone: toneProp,
    legacy: sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useBraidTheme().legacy
  });
  const stylesForVariant = variants[variant][tone];
  const colorContrast = useColorContrast();
  const lightness = useBackgroundLightness();
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    keyboardFocusable ? /* @__PURE__ */ jsxRuntime.jsx(
      FieldOverlay,
      {
        borderRadius: radius,
        variant: "focus",
        onlyVisibleForKeyboardNavigation: true,
        className: styles_lib_components_Button_Button_css_cjs.focusOverlay
      }
    ) : null,
    /* @__PURE__ */ jsxRuntime.jsx(
      FieldOverlay,
      {
        borderRadius: radius,
        background: stylesForVariant.backgroundHover && typeof stylesForVariant.backgroundHover !== "string" ? colorContrast(stylesForVariant.backgroundHover) : stylesForVariant.backgroundHover,
        className: [
          styles_lib_components_Button_Button_css_cjs.hoverOverlay,
          variant !== "solid" && lightness.lightMode === "dark" ? styles_lib_components_Button_Button_css_cjs.invertedBackgroundsLightMode.hover : null,
          variant !== "solid" && lightness.darkMode === "dark" ? styles_lib_components_Button_Button_css_cjs.invertedBackgroundsDarkMode.hover : null
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx(
      FieldOverlay,
      {
        borderRadius: radius,
        background: stylesForVariant.backgroundActive && typeof stylesForVariant.backgroundActive !== "string" ? colorContrast(stylesForVariant.backgroundActive) : stylesForVariant.backgroundActive,
        className: [
          forceActive ? styles_lib_components_Button_Button_css_cjs.forceActive : void 0,
          styles_lib_components_Button_Button_css_cjs.activeOverlay,
          variant !== "solid" && lightness.lightMode === "dark" ? styles_lib_components_Button_Button_css_cjs.invertedBackgroundsLightMode.active : null,
          variant !== "solid" && lightness.darkMode === "dark" ? styles_lib_components_Button_Button_css_cjs.invertedBackgroundsDarkMode.active : null
        ]
      }
    ),
    stylesForVariant.boxShadow ? /* @__PURE__ */ jsxRuntime.jsx(
      Box,
      {
        component: "span",
        boxShadow: stylesForVariant.boxShadow,
        borderRadius: radius,
        position: "absolute",
        inset: 0,
        pointerEvents: "none"
      }
    ) : null
  ] });
};
const ButtonText = ({
  children,
  loading,
  size: sizeProp,
  icon,
  iconPosition = "leading",
  variant: variantProp,
  tone: toneProp,
  bleed
}) => {
  const { variant, tone } = resolveToneAndVariant({
    variant: variantProp,
    tone: toneProp,
    legacy: sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useBraidTheme().legacy
  });
  const lightness = useBackgroundLightness();
  const actionsContext = React.useContext(ActionsContext);
  const isLegacyTheme = sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useBraidTheme().legacy;
  const size = sizeProp ?? (actionsContext == null ? void 0 : actionsContext.size) ?? "standard";
  const stylesForVariant = variants[variant][tone];
  const shouldReducePaddingX = size === "small" || variant === "transparent";
  const labelPaddingXForTheme = isLegacyTheme ? "medium" : "gutter";
  const labelPaddingX = shouldReducePaddingX ? transparentPaddingX : labelPaddingXForTheme;
  assert__default.default(
    !icon || icon.props.size === void 0 && icon.props.tone === void 0,
    "Icons cannot set the 'size' or 'tone' prop when passed to a Button component"
  );
  return /* @__PURE__ */ jsxRuntime.jsx(
    Box,
    {
      component: "span",
      position: "relative",
      display: "flex",
      justifyContent: "center",
      flexWrap: "wrap",
      overflow: "hidden",
      pointerEvents: "none",
      paddingX: labelPaddingX,
      className: styles_lib_components_Button_Button_css_cjs.padToMinHeight,
      background: tone === "neutral" && variant !== "solid" ? {
        lightMode: lightness.lightMode === "light" ? "customLight" : "customDark",
        darkMode: lightness.darkMode === "light" ? "customLight" : "customDark"
      } : void 0,
      children: /* @__PURE__ */ jsxRuntime.jsxs(
        Text,
        {
          tone: stylesForVariant.textTone,
          weight: "medium",
          align: "center",
          size,
          children: [
            icon && iconPosition === "leading" ? /* @__PURE__ */ jsxRuntime.jsx(
              AvoidWidowIcon,
              {
                iconPosition,
                space: "xsmall",
                className: shouldReducePaddingX || bleed ? null : negativeMargin("left", "xxsmall"),
                children: icon
              }
            ) : null,
            children,
            loading ? /* @__PURE__ */ jsxRuntime.jsx(ButtonLoader, {}) : null,
            !loading && icon && iconPosition === "trailing" ? /* @__PURE__ */ jsxRuntime.jsx(
              AvoidWidowIcon,
              {
                iconPosition,
                space: "xsmall",
                className: shouldReducePaddingX || bleed ? null : negativeMargin("right", "xxsmall"),
                children: icon
              }
            ) : null
          ]
        }
      )
    }
  );
};
const useButtonStyles = ({
  variant: variantProp,
  size: sizeProp,
  tone: toneProp,
  loading,
  radius = buttonRadius,
  bleed
}) => {
  const { variant, tone } = resolveToneAndVariant({
    variant: variantProp,
    tone: toneProp,
    legacy: sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useBraidTheme().legacy
  });
  const actionsContext = React.useContext(ActionsContext);
  const size = sizeProp ?? (actionsContext == null ? void 0 : actionsContext.size) ?? "standard";
  const stylesForVariant = variants[variant][tone];
  const colorConstrast = useColorContrast();
  const lightness = useBackgroundLightness();
  return {
    borderRadius: radius,
    width: "full",
    position: "relative",
    display: "block",
    transform: { active: "touchable" },
    transition: "touchable",
    outline: "none",
    textAlign: "center",
    userSelect: "none",
    cursor: !loading ? "pointer" : void 0,
    background: stylesForVariant.background && typeof stylesForVariant.background !== "string" ? colorConstrast(stylesForVariant.background) : stylesForVariant.background,
    className: [
      variant === "soft" && lightness.lightMode === "dark" ? styles_lib_components_Button_Button_css_cjs.invertedBackgroundsLightMode.soft : null,
      variant === "soft" && lightness.darkMode === "dark" ? styles_lib_components_Button_Button_css_cjs.invertedBackgroundsDarkMode.soft : null,
      styles_lib_components_Button_Button_css_cjs.root,
      size === "small" ? virtualTouchable({ xAxis: false }) : null,
      size === "standard" ? styles_lib_components_Button_Button_css_cjs.standard : styles_lib_components_Button_Button_css_cjs.small,
      bleed ? styles_lib_components_Button_Button_css_cjs.bleedVerticallyToCapHeight : null
    ]
  };
};
const ButtonContainer = ({
  children,
  bleed,
  variant = "solid"
}) => bleed && variant === "transparent" ? /* @__PURE__ */ jsxRuntime.jsx(Bleed, { horizontal: transparentPaddingX, children }) : /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, { children });
const Button = React.forwardRef(
  ({
    onClick,
    children,
    size,
    tone,
    icon,
    iconPosition,
    bleedY,
    bleed,
    variant,
    loading,
    type = "button",
    id,
    tabIndex,
    onKeyUp,
    onKeyDown,
    "aria-haspopup": ariaHasPopup,
    "aria-controls": ariaControls,
    "aria-expanded": ariaExpanded,
    "aria-describedby": ariaDescribedBy,
    "aria-label": ariaLabel,
    data,
    ...restProps
  }, ref) => {
    if (process.env.NODE_ENV !== "production") {
      if (typeof bleedY !== "undefined") {
        console.warn(
          dedent__default.default`
            The "bleedY" prop has been deprecated and will be removed in a future version. Use "bleed" instead.
               <Button
              %c-   bleedY
              %c+   bleed
               %c/>
          `,
          "color: red",
          "color: green",
          "color: inherit"
        );
      }
    }
    return /* @__PURE__ */ jsxRuntime.jsx(ButtonContainer, { bleed, variant, children: /* @__PURE__ */ jsxRuntime.jsxs(
      Box,
      {
        component: "button",
        ref,
        id,
        type,
        tabIndex,
        onKeyUp,
        onKeyDown,
        "aria-haspopup": ariaHasPopup,
        "aria-controls": ariaControls,
        "aria-expanded": ariaExpanded,
        "aria-describedby": ariaDescribedBy,
        "aria-label": ariaLabel,
        onClick,
        disabled: loading,
        ...buildDataAttributes({ data, validateRestProps: restProps }),
        ...useButtonStyles({
          variant,
          tone,
          size,
          bleed: bleed || bleedY,
          loading
        }),
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(ButtonOverlays, { variant, tone }),
          /* @__PURE__ */ jsxRuntime.jsx(
            ButtonText,
            {
              variant,
              tone,
              size,
              loading,
              icon,
              iconPosition,
              bleed,
              children
            }
          )
        ]
      }
    ) });
  }
);
Button.displayName = "Button";
const StaticTooltipContext = React.createContext(false);
const TooltipTextDefaultsProvider = ({
  children
}) => {
  const themeName = useThemeName();
  return /* @__PURE__ */ jsxRuntime.jsx(
    DefaultTextPropsProvider,
    {
      size: themeName === "docs" ? "small" : void 0,
      weight: "medium",
      children
    }
  );
};
const borderRadius$1 = "large";
const TooltipContent = ({
  children,
  opacity,
  arrowProps
}) => /* @__PURE__ */ jsxRuntime.jsx(
  Box,
  {
    display: "flex",
    position: "relative",
    transition: "fast",
    opacity: opacity === 0 ? 0 : void 0,
    className: opacity === 0 ? styles_lib_components_TooltipRenderer_TooltipRenderer_css_cjs.verticalOffsetBeforeEntrance : styles_lib_components_TooltipRenderer_TooltipRenderer_css_cjs.translateZ0,
    children: /* @__PURE__ */ jsxRuntime.jsx(
      Box,
      {
        boxShadow: "large",
        background: "neutral",
        borderRadius: borderRadius$1,
        padding: "small",
        className: [styles_lib_components_TooltipRenderer_TooltipRenderer_css_cjs.maxWidth, styles_lib_components_TooltipRenderer_TooltipRenderer_css_cjs.translateZ0],
        children: /* @__PURE__ */ jsxRuntime.jsxs(TooltipTextDefaultsProvider, { children: [
          /* @__PURE__ */ jsxRuntime.jsx(Box, { position: "relative", zIndex: 1, children }),
          /* @__PURE__ */ jsxRuntime.jsx(
            Box,
            {
              ...arrowProps,
              borderRadius: borderRadius$1,
              background: "neutral",
              className: styles_lib_components_TooltipRenderer_TooltipRenderer_css_cjs.arrow
            }
          )
        ] })
      }
    )
  }
);
const validPlacements = ["top", "bottom"];
const normaliseRect = (domRect) => ({
  top: Math.round((domRect == null ? void 0 : domRect.top) || 0),
  left: Math.round((domRect == null ? void 0 : domRect.left) || 0),
  height: Math.round((domRect == null ? void 0 : domRect.height) || 0),
  width: Math.round((domRect == null ? void 0 : domRect.width) || 0)
});
const defaultRect = normaliseRect();
const doesBoundingBoxNeedUpdating = (element, previousRect) => {
  const currentRect = normaliseRect(element == null ? void 0 : element.getBoundingClientRect());
  return JSON.stringify(currentRect) !== JSON.stringify(previousRect);
};
const TooltipRenderer = ({
  id,
  tooltip,
  placement = "top",
  children
}) => {
  assert__default.default(
    validPlacements.includes(placement),
    `The 'placement' prop must be one of the following: ${validPlacements.join(
      ", "
    )}`
  );
  const isStatic = React.useContext(StaticTooltipContext);
  const [controlledVisible, setControlledVisible] = React.useState(false);
  const [opacity, setOpacity] = React.useState(0);
  const { grid, space } = useSpace();
  const triggerBoundingBoxRef = React.useRef(defaultRect);
  const tooltipBoundingRectRef = React.useRef(defaultRect);
  const {
    visible,
    getTooltipProps,
    setTooltipRef,
    tooltipRef,
    setTriggerRef,
    triggerRef,
    getArrowProps,
    update
  } = reactPopperTooltip.usePopperTooltip(
    {
      placement,
      trigger: [isMobile__default.default() ? "click" : "hover", "focus"],
      visible: isStatic || controlledVisible,
      onVisibleChange: (newState) => {
        triggerBoundingBoxRef.current = normaliseRect(
          triggerRef == null ? void 0 : triggerRef.getBoundingClientRect()
        );
        tooltipBoundingRectRef.current = normaliseRect(
          tooltipRef == null ? void 0 : tooltipRef.getBoundingClientRect()
        );
        setControlledVisible(newState);
      }
    },
    {
      modifiers: [
        {
          name: "preventOverflow",
          options: {
            padding: space.xsmall * grid
          }
        },
        {
          name: "offset",
          options: {
            offset: [0, space.small * grid]
          }
        },
        {
          name: "arrow",
          options: {
            padding: space.xsmall * grid
          }
        },
        ...isStatic ? [
          {
            name: "flip",
            options: {
              fallbackPlacements: []
            }
          }
        ] : []
      ]
    }
  );
  React.useLayoutEffect(() => {
    if (controlledVisible && update && (doesBoundingBoxNeedUpdating(triggerRef, triggerBoundingBoxRef.current) || doesBoundingBoxNeedUpdating(tooltipRef, tooltipBoundingRectRef.current))) {
      triggerBoundingBoxRef.current = normaliseRect(
        triggerRef == null ? void 0 : triggerRef.getBoundingClientRect()
      );
      tooltipBoundingRectRef.current = normaliseRect(
        tooltipRef == null ? void 0 : tooltipRef.getBoundingClientRect()
      );
      update();
    }
  });
  React.useEffect(() => {
    if (visible) {
      const handleKeyDown = ({ key }) => {
        if (key === "Escape") {
          setControlledVisible(false);
        }
      };
      const handleScroll = () => {
        setControlledVisible(false);
      };
      const scrollHandlerOptions = {
        capture: true,
        passive: true
      };
      document.addEventListener("keydown", handleKeyDown);
      document.addEventListener("scroll", handleScroll, scrollHandlerOptions);
      return () => {
        document.removeEventListener("keydown", handleKeyDown);
        document.removeEventListener(
          "scroll",
          handleScroll,
          scrollHandlerOptions
        );
      };
    }
  }, [visible]);
  React.useEffect(() => {
    if (!triggerRef) {
      return;
    }
    if (visible) {
      const handleFocusIn = (event) => {
        if (event.currentTarget !== triggerRef) {
          setControlledVisible(false);
        }
      };
      document.addEventListener("focusin", handleFocusIn);
      return () => {
        document.removeEventListener("focusin", handleFocusIn);
      };
    }
  }, [triggerRef, visible]);
  assert__default.default(
    React.useEffect(() => {
      if (tooltipRef) {
        assert__default.default(
          tooltipRef.querySelectorAll("button, input, select, textarea, a").length === 0,
          "For accessibility reasons, tooltips must not contain interactive elements"
        );
      }
    }, [tooltipRef]) === void 0
  );
  React.useEffect(() => {
    if (!tooltipRef || !visible) {
      return setOpacity(0);
    }
    const timeout = setTimeout(() => setOpacity(100), isMobile__default.default() ? 0 : 250);
    return () => clearTimeout(timeout);
  }, [tooltipRef, visible]);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    children({
      triggerProps: {
        tabIndex: 0,
        ref: setTriggerRef,
        "aria-describedby": id
      }
    }),
    triggerRef && /* @__PURE__ */ jsxRuntime.jsx(BraidPortal, { children: /* @__PURE__ */ jsxRuntime.jsx(
      "div",
      {
        id,
        role: "tooltip",
        hidden: !visible ? true : void 0,
        className: styles_lib_css_atoms_atoms_cjs.atoms({
          reset: "div",
          zIndex: "notification",
          pointerEvents: "none",
          display: triggerRef && visible ? void 0 : "none"
        }),
        ...visible ? getTooltipProps({
          ref: setTooltipRef
        }) : null,
        children: /* @__PURE__ */ jsxRuntime.jsx(TooltipContent, { opacity, arrowProps: getArrowProps(), children: tooltip })
      }
    ) })
  ] });
};
const buttonIconVariants = ["soft", "transparent"];
const buttonIconTones = ["neutral", "secondary"];
const padding = "xsmall";
const PrivateButtonIcon = React.forwardRef(
  ({
    icon,
    label,
    id,
    size = "standard",
    tone = "neutral",
    variant = "soft",
    type = "button",
    bleed,
    onClick,
    onMouseDown,
    onKeyUp,
    onKeyDown,
    "aria-haspopup": ariaHasPopUp,
    "aria-expanded": ariaExpanded,
    tabIndex,
    data,
    ...restProps
  }, forwardedRef) => {
    const {
      className: buttonClasses,
      width: _,
      ...buttonStyleProps
    } = useButtonStyles({
      variant,
      tone: "neutral",
      size: "standard",
      radius: "full"
    });
    assert__default.default(
      icon.props.size === void 0,
      "Icons cannot set the 'size' prop when passed to a ButtonIcon component"
    );
    const button = /* @__PURE__ */ jsxRuntime.jsxs(
      Box,
      {
        component: "button",
        type,
        id,
        zIndex: 0,
        title: !id ? label : void 0,
        ref: forwardedRef,
        "aria-label": label,
        "aria-haspopup": ariaHasPopUp,
        "aria-expanded": ariaExpanded,
        padding,
        onClick,
        onKeyUp,
        onKeyDown,
        onMouseDown,
        className: [buttonClasses, styles_lib_components_ButtonIcon_ButtonIcon_css_cjs.button, virtualTouchable()],
        tabIndex,
        ...buildDataAttributes({ data, validateRestProps: restProps }),
        ...buttonStyleProps,
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(
            ButtonOverlays,
            {
              variant,
              tone: "neutral",
              radius: "full",
              keyboardFocusable: typeof tabIndex === "undefined" || tabIndex >= 0,
              forceActive: ariaExpanded === "true" || ariaExpanded === true
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsx(
            Box,
            {
              component: "span",
              display: "block",
              position: "relative",
              className: size === "large" ? iconContainerSize() : iconSize({ crop: true }),
              children: React.cloneElement(icon, { tone: icon.props.tone || tone, size: "fill" })
            }
          )
        ]
      }
    );
    const shouldBleed = typeof bleed === "undefined" && variant === "transparent" || bleed;
    return shouldBleed ? /* @__PURE__ */ jsxRuntime.jsx(Bleed, { space: padding, children: button }) : button;
  }
);
const ButtonIcon = React.forwardRef(
  ({ id, label, ...restProps }, forwardedRef) => {
    if (!id) {
      return /* @__PURE__ */ jsxRuntime.jsx(PrivateButtonIcon, { label, ref: forwardedRef, ...restProps });
    }
    return /* @__PURE__ */ jsxRuntime.jsx(TooltipRenderer, { id: `${id}-tooltip`, tooltip: /* @__PURE__ */ jsxRuntime.jsx(Text, { children: label }), children: ({ triggerProps: { ref: triggerRef, ...triggerProps } }) => /* @__PURE__ */ jsxRuntime.jsx(
      PrivateButtonIcon,
      {
        id,
        label,
        ref: (node) => {
          if (forwardedRef) {
            if (typeof forwardedRef === "function") {
              forwardedRef(node);
            } else {
              forwardedRef.current = node;
            }
          }
          triggerRef(node);
        },
        ...triggerProps,
        ...restProps
      }
    ) });
  }
);
ButtonIcon.displayName = "ButtonIcon";
const ContentBlock = ({
  width = "medium",
  data,
  children,
  ...restProps
}) => /* @__PURE__ */ jsxRuntime.jsx(
  Box,
  {
    width: "full",
    maxWidth: width,
    className: styles_lib_components_ContentBlock_ContentBlock_css_cjs.marginAuto,
    ...buildDataAttributes({ data, validateRestProps: restProps }),
    children
  }
);
const isPlainBackground = (backgroundContext2, contrast) => contrast === "light" && (backgroundContext2 === "body" || backgroundContext2 === "surface" || backgroundContext2 === "neutralLight") || contrast === "dark" && (backgroundContext2 === "bodyDark" || backgroundContext2 === "surfaceDark" || backgroundContext2 === "neutral");
const useLinkStyles = ({
  reset = "a",
  weight,
  showVisited = false,
  hitArea = "standard"
}) => {
  const backgroundLightness = useBackgroundLightness();
  const backgroundContext2 = useBackground();
  const textContext = React.useContext(TextContext);
  const headingContext = React.useContext(HeadingContext);
  const isWeakDueToTextTone = !headingContext && (textContext == null ? void 0 : textContext.tone) !== "neutral" && (textContext == null ? void 0 : textContext.tone) !== "secondary";
  const linkStyles = weight === "weak" || isWeakDueToTextTone ? styles_lib_components_TextLink_TextLink_css_cjs.weakLink : [
    isPlainBackground(backgroundContext2.lightMode, "light") || weight === "regular" ? styles_lib_components_TextLink_TextLink_css_cjs.regularLinkLightMode[backgroundLightness.lightMode] : styles_lib_components_TextLink_TextLink_css_cjs.weakLink,
    isPlainBackground(backgroundContext2.darkMode, "dark") || weight === "regular" ? styles_lib_components_TextLink_TextLink_css_cjs.regularLinkDarkMode[backgroundLightness.darkMode] : styles_lib_components_TextLink_TextLink_css_cjs.weakLink
  ];
  return clsx__default.default(
    styles_lib_components_TextLink_TextLink_css_cjs.base,
    linkStyles,
    weight !== "weak" ? styles_lib_css_typography_css_cjs.fontWeight.medium : void 0,
    showVisited ? [
      styles_lib_components_TextLink_TextLink_css_cjs.visitedLightMode[backgroundLightness.lightMode],
      styles_lib_components_TextLink_TextLink_css_cjs.visitedDarkMode[backgroundLightness.darkMode]
    ] : "",
    reset !== false ? styles_lib_css_atoms_atoms_cjs.atoms({
      reset: typeof reset === "string" ? reset : "a"
    }) : null,
    styles_lib_css_atoms_atoms_cjs.atoms({
      cursor: "pointer"
    }),
    hitArea === "large" && virtualTouchable()
  );
};
const TextLinkContent = ({
  icon,
  iconPosition = "leading",
  children
}) => /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
  icon && iconPosition === "leading" ? /* @__PURE__ */ jsxRuntime.jsx(AvoidWidowIcon, { iconPosition, children: icon }) : null,
  children,
  icon && iconPosition === "trailing" ? /* @__PURE__ */ jsxRuntime.jsx(AvoidWidowIcon, { iconPosition, children: icon }) : null
] });
const TextLink = React.forwardRef(
  ({
    weight,
    showVisited,
    hitArea,
    data,
    icon,
    children,
    iconPosition,
    ...restProps
  }, ref) => {
    const LinkComponent = sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useLinkComponent(ref);
    const classes = useLinkStyles({
      weight,
      showVisited,
      hitArea
    });
    return /* @__PURE__ */ jsxRuntime.jsx(
      LinkComponent,
      {
        ref,
        ...restProps,
        className: classes,
        ...buildDataAttributes({ data, validateRestProps: false }),
        children: /* @__PURE__ */ jsxRuntime.jsx(TextLinkContent, { icon, iconPosition, children })
      }
    );
  }
);
TextLink.displayName = "TextLink";
const noop = () => {
};
const TextLinkButton = ({
  weight,
  hitArea,
  id,
  onClick = noop,
  data,
  children,
  "aria-controls": ariaControls,
  "aria-expanded": ariaExpanded,
  "aria-describedby": ariaDescribedBy,
  "aria-label": ariaLabel,
  tabIndex,
  icon,
  iconPosition,
  ...restProps
}) => {
  const buttonRef = React.useRef(null);
  const classes = useLinkStyles({
    reset: false,
    weight,
    hitArea
  });
  const handleKeyboard = React.useCallback(
    (event) => {
      var _a;
      if (event.key === "Enter" || event.key === " ") {
        event.preventDefault();
        (_a = buttonRef.current) == null ? void 0 : _a.click();
      }
    },
    [buttonRef]
  );
  return /* @__PURE__ */ jsxRuntime.jsx(
    Box,
    {
      ref: buttonRef,
      role: "button",
      tabIndex: tabIndex ?? 0,
      component: "span",
      onClick,
      onKeyDown: handleKeyboard,
      "aria-controls": ariaControls,
      "aria-expanded": ariaExpanded,
      "aria-describedby": ariaDescribedBy,
      "aria-label": ariaLabel,
      id,
      className: classes,
      ...buildDataAttributes({ data, validateRestProps: restProps }),
      children: /* @__PURE__ */ jsxRuntime.jsx(TextLinkContent, { icon, iconPosition, children })
    }
  );
};
const useTimeout = ({ onTimeout, duration }) => {
  const [activated, setActivated] = React.useState(true);
  const timeoutRef = React.useRef();
  const stopTimeout = React.useCallback(() => {
    window.clearTimeout(timeoutRef.current);
    setActivated(false);
  }, []);
  React.useEffect(() => {
    if (activated) {
      timeoutRef.current = window.setTimeout(() => {
        onTimeout();
      }, duration);
      return () => {
        stopTimeout();
      };
    }
  }, [onTimeout, activated, duration, stopTimeout]);
  const startTimeout = React.useCallback(() => {
    setActivated(true);
  }, []);
  return {
    stopTimeout,
    startTimeout
  };
};
const toneToIcon = {
  critical: IconCritical,
  positive: IconPositive
};
const toastDuration = 1e4;
const borderRadius = "large";
const Action = ({ label, onClick, removeToast }) => {
  const handleClick = React.useCallback(() => {
    removeToast();
    onClick();
  }, [removeToast, onClick]);
  return /* @__PURE__ */ jsxRuntime.jsx(Text, { baseline: false, children: /* @__PURE__ */ jsxRuntime.jsx(Box, { component: "span", "aria-hidden": true, children: /* @__PURE__ */ jsxRuntime.jsx(TextLinkButton, { onClick: handleClick, hitArea: "large", children: label }) }) });
};
const ToastIcon = ({ tone, icon }) => {
  if (tone !== "neutral") {
    const Icon = toneToIcon[tone];
    return /* @__PURE__ */ jsxRuntime.jsx(Icon, { tone });
  }
  if (icon) {
    return React.cloneElement(icon, { tone });
  }
  return null;
};
const Toast = React__default.default.forwardRef(
  ({
    id,
    vanillaTheme,
    dedupeKey,
    message,
    description,
    tone,
    icon,
    onClose,
    closeLabel = "Close",
    action,
    shouldRemove,
    data,
    ...restProps
  }, ref) => {
    const remove = React.useCallback(
      () => onClose(dedupeKey, id),
      [onClose, dedupeKey, id]
    );
    const { stopTimeout, startTimeout } = useTimeout({
      duration: toastDuration,
      onTimeout: remove
    });
    React.useEffect(() => {
      if (shouldRemove) {
        stopTimeout();
        remove();
      }
    }, [shouldRemove, remove, stopTimeout]);
    assert__default.default(
      !icon || icon.props.size === void 0 && icon.props.tone === void 0,
      "Icons cannot set the 'size' or 'tone' prop when passed to a Toast component"
    );
    assert__default.default(
      !icon || icon && tone === "neutral",
      `Icons cannot be customised on a Toast component using '${tone}' tone`
    );
    const content = description ? /* @__PURE__ */ jsxRuntime.jsxs(Stack, { space: "xxsmall", children: [
      /* @__PURE__ */ jsxRuntime.jsx(Text, { weight: "medium", tone, baseline: false, children: message }),
      description ? /* @__PURE__ */ jsxRuntime.jsx(Text, { baseline: false, tone: "secondary", children: description }) : null,
      action ? /* @__PURE__ */ jsxRuntime.jsx(Action, { removeToast: remove, ...action }, action.label) : null
    ] }) : /* @__PURE__ */ jsxRuntime.jsxs(Inline, { space: "xxsmall", children: [
      /* @__PURE__ */ jsxRuntime.jsx(Box, { paddingRight: "medium", children: /* @__PURE__ */ jsxRuntime.jsx(Text, { weight: "medium", tone, baseline: false, children: message }) }),
      action ? /* @__PURE__ */ jsxRuntime.jsx(Action, { removeToast: remove, ...action }, action.label) : null
    ] });
    return /* @__PURE__ */ jsxRuntime.jsx(
      Box,
      {
        display: "flex",
        justifyContent: "center",
        role: "alert",
        ref,
        onMouseEnter: stopTimeout,
        onMouseLeave: startTimeout,
        className: vanillaTheme,
        ...buildDataAttributes({ data, validateRestProps: restProps }),
        children: /* @__PURE__ */ jsxRuntime.jsx(Box, { boxShadow: "large", borderRadius, children: /* @__PURE__ */ jsxRuntime.jsx(ContentBlock, { width: "xsmall", children: /* @__PURE__ */ jsxRuntime.jsx(
          Box,
          {
            background: { lightMode: "surfaceDark", darkMode: "surface" },
            position: "relative",
            borderRadius,
            paddingY: "medium",
            paddingX: "gutter",
            overflow: "hidden",
            className: styles_lib_components_useToast_Toast_css_cjs.toast,
            children: /* @__PURE__ */ jsxRuntime.jsxs(Columns, { space: "none", children: [
              tone !== "neutral" || tone === "neutral" && icon ? /* @__PURE__ */ jsxRuntime.jsx(Column, { width: "content", children: /* @__PURE__ */ jsxRuntime.jsx(Box, { paddingRight: "small", children: /* @__PURE__ */ jsxRuntime.jsx(ToastIcon, { tone, icon }) }) }) : null,
              /* @__PURE__ */ jsxRuntime.jsx(Column, { children: content }),
              /* @__PURE__ */ jsxRuntime.jsx(Column, { width: "content", children: /* @__PURE__ */ jsxRuntime.jsx(
                Box,
                {
                  width: "touchable",
                  display: "flex",
                  justifyContent: "flexEnd",
                  alignItems: "center",
                  className: styles_lib_css_lineHeightContainer_css_cjs.lineHeightContainer.standard,
                  "aria-hidden": true,
                  children: /* @__PURE__ */ jsxRuntime.jsx(
                    ButtonIcon,
                    {
                      id: `${dedupeKey}-clear`,
                      icon: /* @__PURE__ */ jsxRuntime.jsx(IconClear, {}),
                      tone: "secondary",
                      variant: "transparent",
                      onClick: remove,
                      label: closeLabel,
                      data: process.env.NODE_ENV !== "production" ? { testid: "clearToast" } : {}
                    }
                  )
                }
              ) })
            ] })
          }
        ) }) })
      }
    );
  }
);
const ToastComponent = Toast;
const animationTimeout = 300;
const entranceTransition = "transform 0.2s ease, opacity 0.2s ease";
const exitTransition = "opacity 0.1s ease";
const animate = (element, transforms, transition, done) => {
  const fallbackTimeout = setTimeout(() => {
    if (done) {
      done();
    }
  }, animationTimeout);
  transforms.forEach(({ property, from = "" }) => {
    element.style.setProperty(property, from);
  });
  element.style.setProperty("transition", "");
  const transitionEndHandler = (ev) => {
    if (ev.target !== element) {
      return;
    }
    element.style.setProperty("transition", "");
    if (done) {
      done();
    }
    element.removeEventListener("transitionend", transitionEndHandler);
    clearTimeout(fallbackTimeout);
  };
  element.addEventListener("transitionend", transitionEndHandler);
  window.requestAnimationFrame(() => {
    window.requestAnimationFrame(() => {
      element.style.setProperty("transition", transition);
      transforms.forEach(({ property, to = "" }) => {
        element.style.setProperty(property, to);
      });
    });
  });
};
const useFlipList = () => {
  const refs = React.useMemo(() => /* @__PURE__ */ new Map(), []);
  const positions = React.useMemo(() => /* @__PURE__ */ new Map(), []);
  React.useLayoutEffect(() => {
    const animations = [];
    Array.from(refs.entries()).forEach(([id, element]) => {
      if (element) {
        const prevTop = positions.get(id);
        const { top, height } = element.getBoundingClientRect();
        if (typeof prevTop === "number" && prevTop !== top) {
          animations.push({
            element,
            transition: entranceTransition,
            transforms: [
              {
                property: "transform",
                from: `translateY(${prevTop - top}px)`
              }
            ]
          });
        } else if (typeof prevTop !== "number") {
          animations.push({
            element,
            transition: entranceTransition,
            transforms: [
              {
                property: "transform",
                from: `translateY(${height}px)`
              },
              {
                property: "opacity",
                from: "0"
              }
            ]
          });
        }
        positions.set(id, element.getBoundingClientRect().top);
      } else {
        refs.delete(id);
      }
    });
    animations.forEach(({ element, transforms, transition }) => {
      animate(element, transforms, transition);
    });
  });
  const remove = React.useCallback(
    (id, cb) => {
      const element = refs.get(id);
      if (element) {
        animate(
          element,
          [
            {
              property: "opacity",
              to: "0"
            }
          ],
          exitTransition,
          cb
        );
      }
    },
    [refs]
  );
  const itemRef = React.useCallback(
    (id) => (ref) => {
      refs.set(id, ref);
    },
    [refs]
  );
  return {
    itemRef,
    remove
  };
};
const Toaster = ({ toasts, removeToast }) => {
  const { itemRef, remove } = useFlipList();
  const onClose = React.useCallback(
    (dedupeKey, id) => {
      remove(id, () => {
        removeToast(dedupeKey);
      });
    },
    [remove, removeToast]
  );
  return /* @__PURE__ */ jsxRuntime.jsx(
    Box,
    {
      position: "fixed",
      zIndex: "notification",
      width: "full",
      pointerEvents: "none",
      paddingX: "gutter",
      bottom: 0,
      children: toasts.map(({ id, ...rest }) => /* @__PURE__ */ jsxRuntime.jsx(Box, { paddingBottom: "small", children: /* @__PURE__ */ jsxRuntime.jsx(
        ToastComponent,
        {
          ref: itemRef(id),
          id,
          onClose,
          ...rest
        }
      ) }, id))
    }
  );
};
let toastCounter = 0;
const ToastControllerContext = React.createContext(null);
const QUEUE_TOAST = 0;
const REMOVE_TOAST = 1;
function reducer(state, action) {
  switch (action.type) {
    case QUEUE_TOAST: {
      const { toast } = action;
      const hasExistingToast = state.toasts.some(
        (t) => t.dedupeKey === toast.dedupeKey
      );
      if (hasExistingToast) {
        return {
          toasts: state.toasts.map((t) => {
            if (t.dedupeKey === toast.dedupeKey) {
              return {
                ...t,
                shouldRemove: true
              };
            }
            return t;
          }),
          queuedToasts: {
            ...state.queuedToasts,
            [toast.dedupeKey]: toast
          }
        };
      }
      return {
        ...state,
        toasts: [...state.toasts, action.toast]
      };
    }
    case REMOVE_TOAST: {
      const toasts = state.toasts.filter(
        ({ dedupeKey }) => dedupeKey !== action.dedupeKey
      );
      const queuedToast = state.queuedToasts[action.dedupeKey];
      if (queuedToast) {
        return {
          queuedToasts: {
            ...state.queuedToasts,
            [action.dedupeKey]: void 0
          },
          toasts: [...toasts, queuedToast]
        };
      }
      return {
        ...state,
        toasts
      };
    }
  }
}
const InternalToastProvider = ({ children }) => {
  const [{ toasts }, dispatch] = React.useReducer(reducer, {
    toasts: [],
    queuedToasts: {}
  });
  const addToast = React.useCallback(
    (toast) => dispatch({ type: QUEUE_TOAST, toast }),
    []
  );
  const removeToast = React.useCallback(
    (dedupeKey) => dispatch({ type: REMOVE_TOAST, dedupeKey }),
    []
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(ToastControllerContext.Provider, { value: addToast, children: [
    children,
    /* @__PURE__ */ jsxRuntime.jsx(ToastPortal, { children: /* @__PURE__ */ jsxRuntime.jsx(Toaster, { toasts, removeToast }) })
  ] });
};
const ToastProvider = ({ children }) => {
  const currentContext = React.useContext(ToastControllerContext);
  if (currentContext !== null) {
    return /* @__PURE__ */ jsxRuntime.jsx(React.Fragment, { children });
  }
  return /* @__PURE__ */ jsxRuntime.jsx(InternalToastProvider, { children });
};
const ToastPortal = ({ children }) => {
  const [toastElement, setElement] = React.useState(null);
  React.useEffect(() => {
    const toastContainerId = "braid-toast-container";
    let element = document.getElementById(toastContainerId);
    if (!element) {
      element = document.createElement("div");
      element.setAttribute("id", toastContainerId);
      element.setAttribute("class", "");
      document.body.appendChild(element);
    }
    setElement(element);
  }, []);
  if (!toastElement) {
    return null;
  }
  return /* @__PURE__ */ jsxRuntime.jsx(BraidPortal, { container: toastElement, children });
};
const useToast = () => {
  const { vanillaTheme } = sideEffects_lib_components_BraidProvider_BraidProvider_cjs.useBraidTheme();
  const addToast = React.useContext(ToastControllerContext);
  if (addToast === null) {
    throw new Error('No "ToastProvider" configured');
  }
  return React.useCallback(
    (toast) => {
      const id = `${toastCounter++}`;
      const dedupeKey = toast.key ?? id;
      addToast({
        ...toast,
        vanillaTheme,
        id,
        dedupeKey,
        shouldRemove: false
      });
    },
    [vanillaTheme, addToast]
  );
};
exports.ActionsContext = ActionsContext;
exports.AvoidWidowIcon = AvoidWidowIcon;
exports.BackgroundProvider = BackgroundProvider;
exports.Bleed = Bleed;
exports.Box = Box;
exports.BraidPortal = BraidPortal;
exports.Button = Button;
exports.ButtonContainer = ButtonContainer;
exports.ButtonIcon = ButtonIcon;
exports.ButtonOverlays = ButtonOverlays;
exports.ButtonText = ButtonText;
exports.Column = Column;
exports.Columns = Columns;
exports.ContentBlock = ContentBlock;
exports.DefaultTextPropsProvider = DefaultTextPropsProvider;
exports.Divider = Divider;
exports.FieldOverlay = FieldOverlay;
exports.HeadingContext = HeadingContext;
exports.Hidden = Hidden;
exports.IconClear = IconClear;
exports.IconCritical = IconCritical;
exports.IconPositive = IconPositive;
exports.IconRenderer = IconRenderer;
exports.Inline = Inline;
exports.Overlay = Overlay;
exports.PublicBox = PublicBox;
exports.Stack = Stack;
exports.Text = Text;
exports.TextContext = TextContext;
exports.TextLink = TextLink;
exports.TextLinkButton = TextLinkButton;
exports.ToastProvider = ToastProvider;
exports.TooltipRenderer = TooltipRenderer;
exports.Typography = Typography;
exports.buildDataAttributes = buildDataAttributes;
exports.buttonIconTones = buttonIconTones;
exports.buttonIconVariants = buttonIconVariants;
exports.buttonTones = buttonTones;
exports.buttonVariants = buttonVariants;
exports.flattenChildren = flattenChildren;
exports.iconContainerSize = iconContainerSize;
exports.iconSize = iconSize;
exports.negativeMargin = negativeMargin;
exports.optimizeResponsiveArray = optimizeResponsiveArray;
exports.resolveResponsiveProp = resolveResponsiveProp;
exports.resolveResponsiveRangeProps = resolveResponsiveRangeProps;
exports.textStyles = textStyles;
exports.useBackground = useBackground;
exports.useBackgroundLightness = useBackgroundLightness;
exports.useButtonStyles = useButtonStyles;
exports.useColoredBoxClasses = useColoredBoxClasses;
exports.useDefaultTextProps = useDefaultTextProps;
exports.useIcon = useIcon;
exports.useResponsiveValue = useResponsiveValue;
exports.useSpace = useSpace;
exports.useThemeName = useThemeName;
exports.useToast = useToast;
exports.validBleedComponents = validBleedComponents;
exports.validColumnsComponents = validColumnsComponents;
exports.validInlineComponents = validInlineComponents;
exports.validStackComponents = validStackComponents;
exports.virtualTouchable = virtualTouchable;
