// braid-design-system/dist/styles/lib/components/private/MaxLines/MaxLines.css.cjs
"use strict";
const fileScope = require("@vanilla-extract/css/fileScope");
const css = require("@vanilla-extract/css");
const styles_entries_css_cjs = require("../../../../../css.cjs");
fileScope.setFileScope("src/lib/components/private/MaxLines/MaxLines.css.ts?used", "braid-design-system");
const base = css.style([styles_entries_css_cjs.atoms({
  display: "block",
  overflow: "hidden",
  minWidth: 0
}), {
  textOverflow: "ellipsis",
  whiteSpace: "nowrap"
}], "base");
const lineLimit = css.createVar("lineLimit");
const lineClampSupportsQuery = "(display: -webkit-box) and (-webkit-line-clamp: 1)";
const multiLine = css.style({
  // Can drop supports query when Edge 16 is removed from browser support policy
  "@supports": {
    [lineClampSupportsQuery]: {
      whiteSpace: "initial",
      display: "-webkit-box",
      WebkitLineClamp: lineLimit,
      WebkitBoxOrient: "vertical"
    }
  }
}, "multiLine");
fileScope.endFileScope();
exports.base = base;
exports.lineLimit = lineLimit;
exports.multiLine = multiLine;
