// braid-design-system/dist/styles/lib/components/private/touchable/virtualTouchable.css.cjs
"use strict";
const fileScope = require("@vanilla-extract/css/fileScope");
const css = require("@vanilla-extract/css");
const styles_lib_components_private_touchable_debugTouchable_cjs = require("./debugTouchable.cjs");
const styles_lib_components_private_touchable_virtualTouchableRules_cjs = require("./virtualTouchableRules.cjs");
fileScope.setFileScope("src/lib/components/private/touchable/virtualTouchable.css.ts?used", "braid-design-system");
const virtualTouchable = css.style([{
  position: "relative",
  ":after": {
    content: '""',
    position: "absolute",
    left: -10,
    right: -10,
    ...styles_lib_components_private_touchable_virtualTouchableRules_cjs.virtualTouchableRules
  }
}, styles_lib_components_private_touchable_debugTouchable_cjs.debugTouchable({
  after: true
})], "virtualTouchable");
const yAxisOnly = css.style({
  ":after": {
    left: "0 !important",
    right: "0 !important"
  }
}, "yAxisOnly");
fileScope.endFileScope();
exports.virtualTouchable = virtualTouchable;
exports.yAxisOnly = yAxisOnly;
